{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "from pandas import Series,DataFrame\n",
    "import requests\n",
    "\n",
    "pd.options.display.max_rows=10\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "a    1\n",
       "b   -2\n",
       "c    4\n",
       "d    9\n",
       "dtype: int64"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "obj = pd.Series([1,-2,4,9],index=['a','b','c','d'])\n",
    "\n",
    "obj"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "data = {'state': ['Ohio', 'Ohio', 'Ohio', 'Nevada', 'Nevada', 'Nevada'],\n",
    "'year': [2000, 2001, 2002, 2001, 2002, 2003],\n",
    "'pop': [1.5, 1.7, 3.6, 2.4, 2.9, 3.2]}\n",
    "\n",
    "dframe = pd.DataFrame(data,columns=[\"year\",\"state\",\"pop\"],)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50181',\n",
       "  'repository_url': 'https://api.github.com/repos/pandas-dev/pandas',\n",
       "  'labels_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50181/labels{/name}',\n",
       "  'comments_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50181/comments',\n",
       "  'events_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50181/events',\n",
       "  'html_url': 'https://github.com/pandas-dev/pandas/pull/50181',\n",
       "  'id': 1489657123,\n",
       "  'node_id': 'PR_kwDOAA0YD85FEZek',\n",
       "  'number': 50181,\n",
       "  'title': 'fix some dosctring-RT02-error',\n",
       "  'user': {'login': 'luke396',\n",
       "   'id': 77875500,\n",
       "   'node_id': 'MDQ6VXNlcjc3ODc1NTAw',\n",
       "   'avatar_url': 'https://avatars.githubusercontent.com/u/77875500?v=4',\n",
       "   'gravatar_id': '',\n",
       "   'url': 'https://api.github.com/users/luke396',\n",
       "   'html_url': 'https://github.com/luke396',\n",
       "   'followers_url': 'https://api.github.com/users/luke396/followers',\n",
       "   'following_url': 'https://api.github.com/users/luke396/following{/other_user}',\n",
       "   'gists_url': 'https://api.github.com/users/luke396/gists{/gist_id}',\n",
       "   'starred_url': 'https://api.github.com/users/luke396/starred{/owner}{/repo}',\n",
       "   'subscriptions_url': 'https://api.github.com/users/luke396/subscriptions',\n",
       "   'organizations_url': 'https://api.github.com/users/luke396/orgs',\n",
       "   'repos_url': 'https://api.github.com/users/luke396/repos',\n",
       "   'events_url': 'https://api.github.com/users/luke396/events{/privacy}',\n",
       "   'received_events_url': 'https://api.github.com/users/luke396/received_events',\n",
       "   'type': 'User',\n",
       "   'site_admin': False},\n",
       "  'labels': [],\n",
       "  'state': 'open',\n",
       "  'locked': False,\n",
       "  'assignee': None,\n",
       "  'assignees': [],\n",
       "  'milestone': None,\n",
       "  'comments': 1,\n",
       "  'created_at': '2022-12-11T08:26:24Z',\n",
       "  'updated_at': '2022-12-11T08:28:50Z',\n",
       "  'closed_at': None,\n",
       "  'author_association': 'NONE',\n",
       "  'active_lock_reason': None,\n",
       "  'draft': False,\n",
       "  'pull_request': {'url': 'https://api.github.com/repos/pandas-dev/pandas/pulls/50181',\n",
       "   'html_url': 'https://github.com/pandas-dev/pandas/pull/50181',\n",
       "   'diff_url': 'https://github.com/pandas-dev/pandas/pull/50181.diff',\n",
       "   'patch_url': 'https://github.com/pandas-dev/pandas/pull/50181.patch',\n",
       "   'merged_at': None},\n",
       "  'body': \"- [x] some #49968 \\r\\n- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature\\r\\n- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).\\r\\n- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.\\r\\n- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.\\r\\n\\r\\nActually I'm not sure they're right, if someone could review and comment this, I will continue to fix the issues.\",\n",
       "  'reactions': {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50181/reactions',\n",
       "   'total_count': 0,\n",
       "   '+1': 0,\n",
       "   '-1': 0,\n",
       "   'laugh': 0,\n",
       "   'hooray': 0,\n",
       "   'confused': 0,\n",
       "   'heart': 0,\n",
       "   'rocket': 0,\n",
       "   'eyes': 0},\n",
       "  'timeline_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50181/timeline',\n",
       "  'performed_via_github_app': None,\n",
       "  'state_reason': None},\n",
       " {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50180',\n",
       "  'repository_url': 'https://api.github.com/repos/pandas-dev/pandas',\n",
       "  'labels_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50180/labels{/name}',\n",
       "  'comments_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50180/comments',\n",
       "  'events_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50180/events',\n",
       "  'html_url': 'https://github.com/pandas-dev/pandas/issues/50180',\n",
       "  'id': 1489581524,\n",
       "  'node_id': 'I_kwDOAA0YD85YyTXU',\n",
       "  'number': 50180,\n",
       "  'title': ' lightfm installation problem',\n",
       "  'user': {'login': 'Abricovich',\n",
       "   'id': 95686247,\n",
       "   'node_id': 'U_kgDOBbQOZw',\n",
       "   'avatar_url': 'https://avatars.githubusercontent.com/u/95686247?v=4',\n",
       "   'gravatar_id': '',\n",
       "   'url': 'https://api.github.com/users/Abricovich',\n",
       "   'html_url': 'https://github.com/Abricovich',\n",
       "   'followers_url': 'https://api.github.com/users/Abricovich/followers',\n",
       "   'following_url': 'https://api.github.com/users/Abricovich/following{/other_user}',\n",
       "   'gists_url': 'https://api.github.com/users/Abricovich/gists{/gist_id}',\n",
       "   'starred_url': 'https://api.github.com/users/Abricovich/starred{/owner}{/repo}',\n",
       "   'subscriptions_url': 'https://api.github.com/users/Abricovich/subscriptions',\n",
       "   'organizations_url': 'https://api.github.com/users/Abricovich/orgs',\n",
       "   'repos_url': 'https://api.github.com/users/Abricovich/repos',\n",
       "   'events_url': 'https://api.github.com/users/Abricovich/events{/privacy}',\n",
       "   'received_events_url': 'https://api.github.com/users/Abricovich/received_events',\n",
       "   'type': 'User',\n",
       "   'site_admin': False},\n",
       "  'labels': [{'id': 129350,\n",
       "    'node_id': 'MDU6TGFiZWwxMjkzNTA=',\n",
       "    'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Build',\n",
       "    'name': 'Build',\n",
       "    'color': '75507B',\n",
       "    'default': False,\n",
       "    'description': 'Library building on various platforms'},\n",
       "   {'id': 1954720290,\n",
       "    'node_id': 'MDU6TGFiZWwxOTU0NzIwMjkw',\n",
       "    'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Needs%20Triage',\n",
       "    'name': 'Needs Triage',\n",
       "    'color': '0052cc',\n",
       "    'default': False,\n",
       "    'description': 'Issue that has not been reviewed by a pandas team member'}],\n",
       "  'state': 'open',\n",
       "  'locked': False,\n",
       "  'assignee': None,\n",
       "  'assignees': [],\n",
       "  'milestone': None,\n",
       "  'comments': 0,\n",
       "  'created_at': '2022-12-11T07:14:51Z',\n",
       "  'updated_at': '2022-12-11T07:14:51Z',\n",
       "  'closed_at': None,\n",
       "  'author_association': 'NONE',\n",
       "  'active_lock_reason': None,\n",
       "  'body': \"### Installation check\\n\\n- [X] I have read the [installation guide](https://pandas.pydata.org/pandas-docs/stable/getting_started/install.html#installing-pandas).\\n\\n\\n### Platform\\n\\nLinux-5.15.0-56-generic-x86_64-with-glibc2.35\\n\\n### Installation Method\\n\\npip install\\n\\n### pandas Version\\n\\n-\\n\\n### Python Version\\n\\n3.10.6\\n\\n### Installation Logs\\n\\n<details>\\r\\n\\r\\nCollecting lightfm\\r\\n  Using cached lightfm-1.16.tar.gz (310 kB)\\r\\n  Preparing metadata (setup.py) ... done\\r\\nRequirement already satisfied: numpy in /usr/local/lib/python3.10/dist-packages (from lightfm) (1.23.5)\\r\\nRequirement already satisfied: scipy>=0.17.0 in /usr/local/lib/python3.10/dist-packages (from lightfm) (1.9.3)\\r\\nRequirement already satisfied: requests in /usr/local/lib/python3.10/dist-packages (from lightfm) (2.28.1)\\r\\nRequirement already satisfied: scikit-learn in /usr/local/lib/python3.10/dist-packages (from lightfm) (1.2.0)\\r\\nRequirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.10/dist-packages (from requests->lightfm) (2022.12.7)\\r\\nRequirement already satisfied: urllib3<1.27,>=1.21.1 in /usr/local/lib/python3.10/dist-packages (from requests->lightfm) (1.26.13)\\r\\nRequirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.10/dist-packages (from requests->lightfm) (3.4)\\r\\nRequirement already satisfied: charset-normalizer<3,>=2 in /usr/local/lib/python3.10/dist-packages (from requests->lightfm) (2.1.1)\\r\\nRequirement already satisfied: joblib>=1.1.1 in /usr/local/lib/python3.10/dist-packages (from scikit-learn->lightfm) (1.2.0)\\r\\nRequirement already satisfied: threadpoolctl>=2.0.0 in /usr/local/lib/python3.10/dist-packages (from scikit-learn->lightfm) (3.1.0)\\r\\nBuilding wheels for collected packages: lightfm\\r\\n  Building wheel for lightfm (setup.py) ... error\\r\\n  error: subprocess-exited-with-error\\r\\n  \\r\\n  × python setup.py bdist_wheel did not run successfully.\\r\\n  │ exit code: 1\\r\\n  ╰─> [176 lines of output]\\r\\n      /usr/local/lib/python3.10/dist-packages/setuptools/dist.py:770: UserWarning: Usage of dash-separated 'description-file' will not be supported in future versions. Please use the underscore name 'description_file' instead\\r\\n        warnings.warn(\\r\\n      running bdist_wheel\\r\\n      running build\\r\\n      running build_py\\r\\n      creating build\\r\\n      creating build/lib.linux-x86_64-cpython-310\\r\\n      creating build/lib.linux-x86_64-cpython-310/lightfm\\r\\n      copying lightfm/cross_validation.py -> build/lib.linux-x86_64-cpython-310/lightfm\\r\\n      copying lightfm/__init__.py -> build/lib.linux-x86_64-cpython-310/lightfm\\r\\n      copying lightfm/lightfm.py -> build/lib.linux-x86_64-cpython-310/lightfm\\r\\n      copying lightfm/data.py -> build/lib.linux-x86_64-cpython-310/lightfm\\r\\n      copying lightfm/_lightfm_fast.py -> build/lib.linux-x86_64-cpython-310/lightfm\\r\\n      copying lightfm/evaluation.py -> build/lib.linux-x86_64-cpython-310/lightfm\\r\\n      creating build/lib.linux-x86_64-cpython-310/lightfm/datasets\\r\\n      copying lightfm/datasets/__init__.py -> build/lib.linux-x86_64-cpython-310/lightfm/datasets\\r\\n      copying lightfm/datasets/movielens.py -> build/lib.linux-x86_64-cpython-310/lightfm/datasets\\r\\n      copying lightfm/datasets/_common.py -> build/lib.linux-x86_64-cpython-310/lightfm/datasets\\r\\n      copying lightfm/datasets/stackexchange.py -> build/lib.linux-x86_64-cpython-310/lightfm/datasets\\r\\n      copying lightfm/_lightfm_fast_no_openmp.c -> build/lib.linux-x86_64-cpython-310/lightfm\\r\\n      copying lightfm/_lightfm_fast_openmp.c -> build/lib.linux-x86_64-cpython-310/lightfm\\r\\n      running build_ext\\r\\n      building 'lightfm._lightfm_fast_openmp' extension\\r\\n      creating build/temp.linux-x86_64-cpython-310\\r\\n      creating build/temp.linux-x86_64-cpython-310/lightfm\\r\\n      x86_64-linux-gnu-gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -fPIC -I/usr/include/python3.10 -c lightfm/_lightfm_fast_openmp.c -o build/temp.linux-x86_64-cpython-310/lightfm/_lightfm_fast_openmp.o -ffast-math -march=native -fopenmp\\r\\n      lightfm/_lightfm_fast_openmp.c: In function ‘__pyx_tp_dealloc_array’:\\r\\n      lightfm/_lightfm_fast_openmp.c:24868:5: error: lvalue required as increment operand\\r\\n      24868 |     ++Py_REFCNT(o);\\r\\n            |     ^~\\r\\n      lightfm/_lightfm_fast_openmp.c:24870:5: error: lvalue required as decrement operand\\r\\n      24870 |     --Py_REFCNT(o);\\r\\n            |     ^~\\r\\n      lightfm/_lightfm_fast_openmp.c: In function ‘__pyx_tp_dealloc_memoryview’:\\r\\n      lightfm/_lightfm_fast_openmp.c:25179:5: error: lvalue required as increment operand\\r\\n      25179 |     ++Py_REFCNT(o);\\r\\n            |     ^~\\r\\n      lightfm/_lightfm_fast_openmp.c:25181:5: error: lvalue required as decrement operand\\r\\n      25181 |     --Py_REFCNT(o);\\r\\n            |     ^~\\r\\n      lightfm/_lightfm_fast_openmp.c: In function ‘__pyx_tp_dealloc__memoryviewslice’:\\r\\n      lightfm/_lightfm_fast_openmp.c:25429:5: error: lvalue required as increment operand\\r\\n      25429 |     ++Py_REFCNT(o);\\r\\n            |     ^~\\r\\n      lightfm/_lightfm_fast_openmp.c:25431:5: error: lvalue required as decrement operand\\r\\n      25431 |     --Py_REFCNT(o);\\r\\n            |     ^~\\r\\n      lightfm/_lightfm_fast_openmp.c: In function ‘__Pyx_InitGlobals’:\\r\\n      lightfm/_lightfm_fast_openmp.c:26184:1: warning: ‘PyEval_InitThreads’ is deprecated [-Wdeprecated-declarations]\\r\\n      26184 | PyEval_InitThreads();\\r\\n            | ^~~~~~~~~~~~~~~~~~\\r\\n      In file included from /usr/include/python3.10/Python.h:130,\\r\\n                       from lightfm/_lightfm_fast_openmp.c:20:\\r\\n      /usr/include/python3.10/ceval.h:122:37: note: declared here\\r\\n        122 | Py_DEPRECATED(3.9) PyAPI_FUNC(void) PyEval_InitThreads(void);\\r\\n            |                                     ^~~~~~~~~~~~~~~~~~\\r\\n      lightfm/_lightfm_fast_openmp.c: In function ‘__Pyx_ParseOptionalKeywords’:\\r\\n      lightfm/_lightfm_fast_openmp.c:26990:21: warning: ‘_PyUnicode_get_wstr_length’ is deprecated [-Wdeprecated-declarations]\\r\\n      26990 |                     (PyUnicode_GET_SIZE(**name) != PyUnicode_GET_SIZE(key)) ? 1 :\\r\\n            |                     ^\\r\\n      In file included from /usr/include/python3.10/unicodeobject.h:1046,\\r\\n                       from /usr/include/python3.10/Python.h:83,\\r\\n                       from lightfm/_lightfm_fast_openmp.c:20:\\r\\n      /usr/include/python3.10/cpython/unicodeobject.h:446:26: note: declared here\\r\\n        446 | static inline Py_ssize_t _PyUnicode_get_wstr_length(PyObject *op) {\\r\\n            |                          ^~~~~~~~~~~~~~~~~~~~~~~~~~\\r\\n      lightfm/_lightfm_fast_openmp.c:26990:21: warning: ‘PyUnicode_AsUnicode’ is deprecated [-Wdeprecated-declarations]\\r\\n      26990 |                     (PyUnicode_GET_SIZE(**name) != PyUnicode_GET_SIZE(key)) ? 1 :\\r\\n            |                     ^\\r\\n      In file included from /usr/include/python3.10/unicodeobject.h:1046,\\r\\n                       from /usr/include/python3.10/Python.h:83,\\r\\n                       from lightfm/_lightfm_fast_openmp.c:20:\\r\\n      /usr/include/python3.10/cpython/unicodeobject.h:580:45: note: declared here\\r\\n        580 | Py_DEPRECATED(3.3) PyAPI_FUNC(Py_UNICODE *) PyUnicode_AsUnicode(\\r\\n            |                                             ^~~~~~~~~~~~~~~~~~~\\r\\n      lightfm/_lightfm_fast_openmp.c:26990:21: warning: ‘_PyUnicode_get_wstr_length’ is deprecated [-Wdeprecated-declarations]\\r\\n      26990 |                     (PyUnicode_GET_SIZE(**name) != PyUnicode_GET_SIZE(key)) ? 1 :\\r\\n            |                     ^\\r\\n      In file included from /usr/include/python3.10/unicodeobject.h:1046,\\r\\n                       from /usr/include/python3.10/Python.h:83,\\r\\n                       from lightfm/_lightfm_fast_openmp.c:20:\\r\\n      /usr/include/python3.10/cpython/unicodeobject.h:446:26: note: declared here\\r\\n        446 | static inline Py_ssize_t _PyUnicode_get_wstr_length(PyObject *op) {\\r\\n            |                          ^~~~~~~~~~~~~~~~~~~~~~~~~~\\r\\n      lightfm/_lightfm_fast_openmp.c:26990:21: warning: ‘_PyUnicode_get_wstr_length’ is deprecated [-Wdeprecated-declarations]\\r\\n      26990 |                     (PyUnicode_GET_SIZE(**name) != PyUnicode_GET_SIZE(key)) ? 1 :\\r\\n            |                     ^\\r\\n      In file included from /usr/include/python3.10/unicodeobject.h:1046,\\r\\n                       from /usr/include/python3.10/Python.h:83,\\r\\n                       from lightfm/_lightfm_fast_openmp.c:20:\\r\\n      /usr/include/python3.10/cpython/unicodeobject.h:446:26: note: declared here\\r\\n        446 | static inline Py_ssize_t _PyUnicode_get_wstr_length(PyObject *op) {\\r\\n            |                          ^~~~~~~~~~~~~~~~~~~~~~~~~~\\r\\n      lightfm/_lightfm_fast_openmp.c:26990:21: warning: ‘PyUnicode_AsUnicode’ is deprecated [-Wdeprecated-declarations]\\r\\n      26990 |                     (PyUnicode_GET_SIZE(**name) != PyUnicode_GET_SIZE(key)) ? 1 :\\r\\n            |                     ^\\r\\n      In file included from /usr/include/python3.10/unicodeobject.h:1046,\\r\\n                       from /usr/include/python3.10/Python.h:83,\\r\\n                       from lightfm/_lightfm_fast_openmp.c:20:\\r\\n      /usr/include/python3.10/cpython/unicodeobject.h:580:45: note: declared here\\r\\n        580 | Py_DEPRECATED(3.3) PyAPI_FUNC(Py_UNICODE *) PyUnicode_AsUnicode(\\r\\n            |                                             ^~~~~~~~~~~~~~~~~~~\\r\\n      lightfm/_lightfm_fast_openmp.c:26990:21: warning: ‘_PyUnicode_get_wstr_length’ is deprecated [-Wdeprecated-declarations]\\r\\n      26990 |                     (PyUnicode_GET_SIZE(**name) != PyUnicode_GET_SIZE(key)) ? 1 :\\r\\n            |                     ^\\r\\n      In file included from /usr/include/python3.10/unicodeobject.h:1046,\\r\\n                       from /usr/include/python3.10/Python.h:83,\\r\\n                       from lightfm/_lightfm_fast_openmp.c:20:\\r\\n      /usr/include/python3.10/cpython/unicodeobject.h:446:26: note: declared here\\r\\n        446 | static inline Py_ssize_t _PyUnicode_get_wstr_length(PyObject *op) {\\r\\n            |                          ^~~~~~~~~~~~~~~~~~~~~~~~~~\\r\\n      lightfm/_lightfm_fast_openmp.c:27006:25: warning: ‘_PyUnicode_get_wstr_length’ is deprecated [-Wdeprecated-declarations]\\r\\n      27006 |                         (PyUnicode_GET_SIZE(**argname) != PyUnicode_GET_SIZE(key)) ? 1 :\\r\\n            |                         ^\\r\\n      In file included from /usr/include/python3.10/unicodeobject.h:1046,\\r\\n                       from /usr/include/python3.10/Python.h:83,\\r\\n                       from lightfm/_lightfm_fast_openmp.c:20:\\r\\n      /usr/include/python3.10/cpython/unicodeobject.h:446:26: note: declared here\\r\\n        446 | static inline Py_ssize_t _PyUnicode_get_wstr_length(PyObject *op) {\\r\\n            |                          ^~~~~~~~~~~~~~~~~~~~~~~~~~\\r\\n      lightfm/_lightfm_fast_openmp.c:27006:25: warning: ‘PyUnicode_AsUnicode’ is deprecated [-Wdeprecated-declarations]\\r\\n      27006 |                         (PyUnicode_GET_SIZE(**argname) != PyUnicode_GET_SIZE(key)) ? 1 :\\r\\n            |                         ^\\r\\n      In file included from /usr/include/python3.10/unicodeobject.h:1046,\\r\\n                       from /usr/include/python3.10/Python.h:83,\\r\\n                       from lightfm/_lightfm_fast_openmp.c:20:\\r\\n      /usr/include/python3.10/cpython/unicodeobject.h:580:45: note: declared here\\r\\n        580 | Py_DEPRECATED(3.3) PyAPI_FUNC(Py_UNICODE *) PyUnicode_AsUnicode(\\r\\n            |                                             ^~~~~~~~~~~~~~~~~~~\\r\\n      lightfm/_lightfm_fast_openmp.c:27006:25: warning: ‘_PyUnicode_get_wstr_length’ is deprecated [-Wdeprecated-declarations]\\r\\n      27006 |                         (PyUnicode_GET_SIZE(**argname) != PyUnicode_GET_SIZE(key)) ? 1 :\\r\\n            |                         ^\\r\\n      In file included from /usr/include/python3.10/unicodeobject.h:1046,\\r\\n                       from /usr/include/python3.10/Python.h:83,\\r\\n                       from lightfm/_lightfm_fast_openmp.c:20:\\r\\n      /usr/include/python3.10/cpython/unicodeobject.h:446:26: note: declared here\\r\\n        446 | static inline Py_ssize_t _PyUnicode_get_wstr_length(PyObject *op) {\\r\\n            |                          ^~~~~~~~~~~~~~~~~~~~~~~~~~\\r\\n      lightfm/_lightfm_fast_openmp.c:27006:25: warning: ‘_PyUnicode_get_wstr_length’ is deprecated [-Wdeprecated-declarations]\\r\\n      27006 |                         (PyUnicode_GET_SIZE(**argname) != PyUnicode_GET_SIZE(key)) ? 1 :\\r\\n            |                         ^\\r\\n      In file included from /usr/include/python3.10/unicodeobject.h:1046,\\r\\n                       from /usr/include/python3.10/Python.h:83,\\r\\n                       from lightfm/_lightfm_fast_openmp.c:20:\\r\\n      /usr/include/python3.10/cpython/unicodeobject.h:446:26: note: declared here\\r\\n        446 | static inline Py_ssize_t _PyUnicode_get_wstr_length(PyObject *op) {\\r\\n            |                          ^~~~~~~~~~~~~~~~~~~~~~~~~~\\r\\n      lightfm/_lightfm_fast_openmp.c:27006:25: warning: ‘PyUnicode_AsUnicode’ is deprecated [-Wdeprecated-declarations]\\r\\n      27006 |                         (PyUnicode_GET_SIZE(**argname) != PyUnicode_GET_SIZE(key)) ? 1 :\\r\\n            |                         ^\\r\\n      In file included from /usr/include/python3.10/unicodeobject.h:1046,\\r\\n                       from /usr/include/python3.10/Python.h:83,\\r\\n                       from lightfm/_lightfm_fast_openmp.c:20:\\r\\n      /usr/include/python3.10/cpython/unicodeobject.h:580:45: note: declared here\\r\\n        580 | Py_DEPRECATED(3.3) PyAPI_FUNC(Py_UNICODE *) PyUnicode_AsUnicode(\\r\\n            |                                             ^~~~~~~~~~~~~~~~~~~\\r\\n      lightfm/_lightfm_fast_openmp.c:27006:25: warning: ‘_PyUnicode_get_wstr_length’ is deprecated [-Wdeprecated-declarations]\\r\\n      27006 |                         (PyUnicode_GET_SIZE(**argname) != PyUnicode_GET_SIZE(key)) ? 1 :\\r\\n            |                         ^\\r\\n      In file included from /usr/include/python3.10/unicodeobject.h:1046,\\r\\n                       from /usr/include/python3.10/Python.h:83,\\r\\n                       from lightfm/_lightfm_fast_openmp.c:20:\\r\\n      /usr/include/python3.10/cpython/unicodeobject.h:446:26: note: declared here\\r\\n        446 | static inline Py_ssize_t _PyUnicode_get_wstr_length(PyObject *op) {\\r\\n            |                          ^~~~~~~~~~~~~~~~~~~~~~~~~~\\r\\n      lightfm/_lightfm_fast_openmp.c: In function ‘__Pyx_decode_c_string’:\\r\\n      lightfm/_lightfm_fast_openmp.c:28218:9: warning: ‘PyUnicode_FromUnicode’ is deprecated [-Wdeprecated-declarations]\\r\\n      28218 |         return PyUnicode_FromUnicode(NULL, 0);\\r\\n            |         ^~~~~~\\r\\n      In file included from /usr/include/python3.10/unicodeobject.h:1046,\\r\\n                       from /usr/include/python3.10/Python.h:83,\\r\\n                       from lightfm/_lightfm_fast_openmp.c:20:\\r\\n      /usr/include/python3.10/cpython/unicodeobject.h:551:42: note: declared here\\r\\n        551 | Py_DEPRECATED(3.3) PyAPI_FUNC(PyObject*) PyUnicode_FromUnicode(\\r\\n            |                                          ^~~~~~~~~~~~~~~~~~~~~\\r\\n      error: command '/usr/bin/x86_64-linux-gnu-gcc' failed with exit code 1\\r\\n      [end of output]\\r\\n  \\r\\n  note: This error originates from a subprocess, and is likely not a problem with pip.\\r\\n  ERROR: Failed building wheel for lightfm\\r\\n  Running setup.py clean for lightfm\\r\\nFailed to build lightfm\\r\\nInstalling collected packages: lightfm\\r\\n  Running setup.py install for lightfm ... error\\r\\n  error: subprocess-exited-with-error\\r\\n  \\r\\n  × Running setup.py install for lightfm did not run successfully.\\r\\n  │ exit code: 1\\r\\n  ╰─> [178 lines of output]\\r\\n      /usr/local/lib/python3.10/dist-packages/setuptools/dist.py:770: UserWarning: Usage of dash-separated 'description-file' will not be supported in future versions. Please use the underscore name 'description_file' instead\\r\\n        warnings.warn(\\r\\n      running install\\r\\n      /usr/local/lib/python3.10/dist-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.\\r\\n        warnings.warn(\\r\\n      running build\\r\\n      running build_py\\r\\n      creating build\\r\\n      creating build/lib.linux-x86_64-cpython-310\\r\\n      creating build/lib.linux-x86_64-cpython-310/lightfm\\r\\n      copying lightfm/cross_validation.py -> build/lib.linux-x86_64-cpython-310/lightfm\\r\\n      copying lightfm/__init__.py -> build/lib.linux-x86_64-cpython-310/lightfm\\r\\n      copying lightfm/lightfm.py -> build/lib.linux-x86_64-cpython-310/lightfm\\r\\n      copying lightfm/data.py -> build/lib.linux-x86_64-cpython-310/lightfm\\r\\n      copying lightfm/_lightfm_fast.py -> build/lib.linux-x86_64-cpython-310/lightfm\\r\\n      copying lightfm/evaluation.py -> build/lib.linux-x86_64-cpython-310/lightfm\\r\\n      creating build/lib.linux-x86_64-cpython-310/lightfm/datasets\\r\\n      copying lightfm/datasets/__init__.py -> build/lib.linux-x86_64-cpython-310/lightfm/datasets\\r\\n      copying lightfm/datasets/movielens.py -> build/lib.linux-x86_64-cpython-310/lightfm/datasets\\r\\n      copying lightfm/datasets/_common.py -> build/lib.linux-x86_64-cpython-310/lightfm/datasets\\r\\n      copying lightfm/datasets/stackexchange.py -> build/lib.linux-x86_64-cpython-310/lightfm/datasets\\r\\n      copying lightfm/_lightfm_fast_no_openmp.c -> build/lib.linux-x86_64-cpython-310/lightfm\\r\\n      copying lightfm/_lightfm_fast_openmp.c -> build/lib.linux-x86_64-cpython-310/lightfm\\r\\n      running build_ext\\r\\n      building 'lightfm._lightfm_fast_openmp' extension\\r\\n      creating build/temp.linux-x86_64-cpython-310\\r\\n      creating build/temp.linux-x86_64-cpython-310/lightfm\\r\\n      x86_64-linux-gnu-gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -fPIC -I/usr/include/python3.10 -c lightfm/_lightfm_fast_openmp.c -o build/temp.linux-x86_64-cpython-310/lightfm/_lightfm_fast_openmp.o -ffast-math -march=native -fopenmp\\r\\n      lightfm/_lightfm_fast_openmp.c: In function ‘__pyx_tp_dealloc_array’:\\r\\n      lightfm/_lightfm_fast_openmp.c:24868:5: error: lvalue required as increment operand\\r\\n      24868 |     ++Py_REFCNT(o);\\r\\n            |     ^~\\r\\n      lightfm/_lightfm_fast_openmp.c:24870:5: error: lvalue required as decrement operand\\r\\n      24870 |     --Py_REFCNT(o);\\r\\n            |     ^~\\r\\n      lightfm/_lightfm_fast_openmp.c: In function ‘__pyx_tp_dealloc_memoryview’:\\r\\n      lightfm/_lightfm_fast_openmp.c:25179:5: error: lvalue required as increment operand\\r\\n      25179 |     ++Py_REFCNT(o);\\r\\n            |     ^~\\r\\n      lightfm/_lightfm_fast_openmp.c:25181:5: error: lvalue required as decrement operand\\r\\n      25181 |     --Py_REFCNT(o);\\r\\n            |     ^~\\r\\n      lightfm/_lightfm_fast_openmp.c: In function ‘__pyx_tp_dealloc__memoryviewslice’:\\r\\n      lightfm/_lightfm_fast_openmp.c:25429:5: error: lvalue required as increment operand\\r\\n      25429 |     ++Py_REFCNT(o);\\r\\n            |     ^~\\r\\n      lightfm/_lightfm_fast_openmp.c:25431:5: error: lvalue required as decrement operand\\r\\n      25431 |     --Py_REFCNT(o);\\r\\n            |     ^~\\r\\n      lightfm/_lightfm_fast_openmp.c: In function ‘__Pyx_InitGlobals’:\\r\\n      lightfm/_lightfm_fast_openmp.c:26184:1: warning: ‘PyEval_InitThreads’ is deprecated [-Wdeprecated-declarations]\\r\\n      26184 | PyEval_InitThreads();\\r\\n            | ^~~~~~~~~~~~~~~~~~\\r\\n      In file included from /usr/include/python3.10/Python.h:130,\\r\\n                       from lightfm/_lightfm_fast_openmp.c:20:\\r\\n      /usr/include/python3.10/ceval.h:122:37: note: declared here\\r\\n        122 | Py_DEPRECATED(3.9) PyAPI_FUNC(void) PyEval_InitThreads(void);\\r\\n            |                                     ^~~~~~~~~~~~~~~~~~\\r\\n      lightfm/_lightfm_fast_openmp.c: In function ‘__Pyx_ParseOptionalKeywords’:\\r\\n      lightfm/_lightfm_fast_openmp.c:26990:21: warning: ‘_PyUnicode_get_wstr_length’ is deprecated [-Wdeprecated-declarations]\\r\\n      26990 |                     (PyUnicode_GET_SIZE(**name) != PyUnicode_GET_SIZE(key)) ? 1 :\\r\\n            |                     ^\\r\\n      In file included from /usr/include/python3.10/unicodeobject.h:1046,\\r\\n                       from /usr/include/python3.10/Python.h:83,\\r\\n                       from lightfm/_lightfm_fast_openmp.c:20:\\r\\n      /usr/include/python3.10/cpython/unicodeobject.h:446:26: note: declared here\\r\\n        446 | static inline Py_ssize_t _PyUnicode_get_wstr_length(PyObject *op) {\\r\\n            |                          ^~~~~~~~~~~~~~~~~~~~~~~~~~\\r\\n      lightfm/_lightfm_fast_openmp.c:26990:21: warning: ‘PyUnicode_AsUnicode’ is deprecated [-Wdeprecated-declarations]\\r\\n      26990 |                     (PyUnicode_GET_SIZE(**name) != PyUnicode_GET_SIZE(key)) ? 1 :\\r\\n            |                     ^\\r\\n      In file included from /usr/include/python3.10/unicodeobject.h:1046,\\r\\n                       from /usr/include/python3.10/Python.h:83,\\r\\n                       from lightfm/_lightfm_fast_openmp.c:20:\\r\\n      /usr/include/python3.10/cpython/unicodeobject.h:580:45: note: declared here\\r\\n        580 | Py_DEPRECATED(3.3) PyAPI_FUNC(Py_UNICODE *) PyUnicode_AsUnicode(\\r\\n            |                                             ^~~~~~~~~~~~~~~~~~~\\r\\n      lightfm/_lightfm_fast_openmp.c:26990:21: warning: ‘_PyUnicode_get_wstr_length’ is deprecated [-Wdeprecated-declarations]\\r\\n      26990 |                     (PyUnicode_GET_SIZE(**name) != PyUnicode_GET_SIZE(key)) ? 1 :\\r\\n            |                     ^\\r\\n      In file included from /usr/include/python3.10/unicodeobject.h:1046,\\r\\n                       from /usr/include/python3.10/Python.h:83,\\r\\n                       from lightfm/_lightfm_fast_openmp.c:20:\\r\\n      /usr/include/python3.10/cpython/unicodeobject.h:446:26: note: declared here\\r\\n        446 | static inline Py_ssize_t _PyUnicode_get_wstr_length(PyObject *op) {\\r\\n            |                          ^~~~~~~~~~~~~~~~~~~~~~~~~~\\r\\n      lightfm/_lightfm_fast_openmp.c:26990:21: warning: ‘_PyUnicode_get_wstr_length’ is deprecated [-Wdeprecated-declarations]\\r\\n      26990 |                     (PyUnicode_GET_SIZE(**name) != PyUnicode_GET_SIZE(key)) ? 1 :\\r\\n            |                     ^\\r\\n      In file included from /usr/include/python3.10/unicodeobject.h:1046,\\r\\n                       from /usr/include/python3.10/Python.h:83,\\r\\n                       from lightfm/_lightfm_fast_openmp.c:20:\\r\\n      /usr/include/python3.10/cpython/unicodeobject.h:446:26: note: declared here\\r\\n        446 | static inline Py_ssize_t _PyUnicode_get_wstr_length(PyObject *op) {\\r\\n            |                          ^~~~~~~~~~~~~~~~~~~~~~~~~~\\r\\n      lightfm/_lightfm_fast_openmp.c:26990:21: warning: ‘PyUnicode_AsUnicode’ is deprecated [-Wdeprecated-declarations]\\r\\n      26990 |                     (PyUnicode_GET_SIZE(**name) != PyUnicode_GET_SIZE(key)) ? 1 :\\r\\n            |                     ^\\r\\n      In file included from /usr/include/python3.10/unicodeobject.h:1046,\\r\\n                       from /usr/include/python3.10/Python.h:83,\\r\\n                       from lightfm/_lightfm_fast_openmp.c:20:\\r\\n      /usr/include/python3.10/cpython/unicodeobject.h:580:45: note: declared here\\r\\n        580 | Py_DEPRECATED(3.3) PyAPI_FUNC(Py_UNICODE *) PyUnicode_AsUnicode(\\r\\n            |                                             ^~~~~~~~~~~~~~~~~~~\\r\\n      lightfm/_lightfm_fast_openmp.c:26990:21: warning: ‘_PyUnicode_get_wstr_length’ is deprecated [-Wdeprecated-declarations]\\r\\n      26990 |                     (PyUnicode_GET_SIZE(**name) != PyUnicode_GET_SIZE(key)) ? 1 :\\r\\n            |                     ^\\r\\n      In file included from /usr/include/python3.10/unicodeobject.h:1046,\\r\\n                       from /usr/include/python3.10/Python.h:83,\\r\\n                       from lightfm/_lightfm_fast_openmp.c:20:\\r\\n      /usr/include/python3.10/cpython/unicodeobject.h:446:26: note: declared here\\r\\n        446 | static inline Py_ssize_t _PyUnicode_get_wstr_length(PyObject *op) {\\r\\n            |                          ^~~~~~~~~~~~~~~~~~~~~~~~~~\\r\\n      lightfm/_lightfm_fast_openmp.c:27006:25: warning: ‘_PyUnicode_get_wstr_length’ is deprecated [-Wdeprecated-declarations]\\r\\n      27006 |                         (PyUnicode_GET_SIZE(**argname) != PyUnicode_GET_SIZE(key)) ? 1 :\\r\\n            |                         ^\\r\\n      In file included from /usr/include/python3.10/unicodeobject.h:1046,\\r\\n                       from /usr/include/python3.10/Python.h:83,\\r\\n                       from lightfm/_lightfm_fast_openmp.c:20:\\r\\n      /usr/include/python3.10/cpython/unicodeobject.h:446:26: note: declared here\\r\\n        446 | static inline Py_ssize_t _PyUnicode_get_wstr_length(PyObject *op) {\\r\\n            |                          ^~~~~~~~~~~~~~~~~~~~~~~~~~\\r\\n      lightfm/_lightfm_fast_openmp.c:27006:25: warning: ‘PyUnicode_AsUnicode’ is deprecated [-Wdeprecated-declarations]\\r\\n      27006 |                         (PyUnicode_GET_SIZE(**argname) != PyUnicode_GET_SIZE(key)) ? 1 :\\r\\n            |                         ^\\r\\n      In file included from /usr/include/python3.10/unicodeobject.h:1046,\\r\\n                       from /usr/include/python3.10/Python.h:83,\\r\\n                       from lightfm/_lightfm_fast_openmp.c:20:\\r\\n      /usr/include/python3.10/cpython/unicodeobject.h:580:45: note: declared here\\r\\n        580 | Py_DEPRECATED(3.3) PyAPI_FUNC(Py_UNICODE *) PyUnicode_AsUnicode(\\r\\n            |                                             ^~~~~~~~~~~~~~~~~~~\\r\\n      lightfm/_lightfm_fast_openmp.c:27006:25: warning: ‘_PyUnicode_get_wstr_length’ is deprecated [-Wdeprecated-declarations]\\r\\n      27006 |                         (PyUnicode_GET_SIZE(**argname) != PyUnicode_GET_SIZE(key)) ? 1 :\\r\\n            |                         ^\\r\\n      In file included from /usr/include/python3.10/unicodeobject.h:1046,\\r\\n                       from /usr/include/python3.10/Python.h:83,\\r\\n                       from lightfm/_lightfm_fast_openmp.c:20:\\r\\n      /usr/include/python3.10/cpython/unicodeobject.h:446:26: note: declared here\\r\\n        446 | static inline Py_ssize_t _PyUnicode_get_wstr_length(PyObject *op) {\\r\\n            |                          ^~~~~~~~~~~~~~~~~~~~~~~~~~\\r\\n      lightfm/_lightfm_fast_openmp.c:27006:25: warning: ‘_PyUnicode_get_wstr_length’ is deprecated [-Wdeprecated-declarations]\\r\\n      27006 |                         (PyUnicode_GET_SIZE(**argname) != PyUnicode_GET_SIZE(key)) ? 1 :\\r\\n            |                         ^\\r\\n      In file included from /usr/include/python3.10/unicodeobject.h:1046,\\r\\n                       from /usr/include/python3.10/Python.h:83,\\r\\n                       from lightfm/_lightfm_fast_openmp.c:20:\\r\\n      /usr/include/python3.10/cpython/unicodeobject.h:446:26: note: declared here\\r\\n        446 | static inline Py_ssize_t _PyUnicode_get_wstr_length(PyObject *op) {\\r\\n            |                          ^~~~~~~~~~~~~~~~~~~~~~~~~~\\r\\n      lightfm/_lightfm_fast_openmp.c:27006:25: warning: ‘PyUnicode_AsUnicode’ is deprecated [-Wdeprecated-declarations]\\r\\n      27006 |                         (PyUnicode_GET_SIZE(**argname) != PyUnicode_GET_SIZE(key)) ? 1 :\\r\\n            |                         ^\\r\\n      In file included from /usr/include/python3.10/unicodeobject.h:1046,\\r\\n                       from /usr/include/python3.10/Python.h:83,\\r\\n                       from lightfm/_lightfm_fast_openmp.c:20:\\r\\n      /usr/include/python3.10/cpython/unicodeobject.h:580:45: note: declared here\\r\\n        580 | Py_DEPRECATED(3.3) PyAPI_FUNC(Py_UNICODE *) PyUnicode_AsUnicode(\\r\\n            |                                             ^~~~~~~~~~~~~~~~~~~\\r\\n      lightfm/_lightfm_fast_openmp.c:27006:25: warning: ‘_PyUnicode_get_wstr_length’ is deprecated [-Wdeprecated-declarations]\\r\\n      27006 |                         (PyUnicode_GET_SIZE(**argname) != PyUnicode_GET_SIZE(key)) ? 1 :\\r\\n            |                         ^\\r\\n      In file included from /usr/include/python3.10/unicodeobject.h:1046,\\r\\n                       from /usr/include/python3.10/Python.h:83,\\r\\n                       from lightfm/_lightfm_fast_openmp.c:20:\\r\\n      /usr/include/python3.10/cpython/unicodeobject.h:446:26: note: declared here\\r\\n        446 | static inline Py_ssize_t _PyUnicode_get_wstr_length(PyObject *op) {\\r\\n            |                          ^~~~~~~~~~~~~~~~~~~~~~~~~~\\r\\n      lightfm/_lightfm_fast_openmp.c: In function ‘__Pyx_decode_c_string’:\\r\\n      lightfm/_lightfm_fast_openmp.c:28218:9: warning: ‘PyUnicode_FromUnicode’ is deprecated [-Wdeprecated-declarations]\\r\\n      28218 |         return PyUnicode_FromUnicode(NULL, 0);\\r\\n            |         ^~~~~~\\r\\n      In file included from /usr/include/python3.10/unicodeobject.h:1046,\\r\\n                       from /usr/include/python3.10/Python.h:83,\\r\\n                       from lightfm/_lightfm_fast_openmp.c:20:\\r\\n      /usr/include/python3.10/cpython/unicodeobject.h:551:42: note: declared here\\r\\n        551 | Py_DEPRECATED(3.3) PyAPI_FUNC(PyObject*) PyUnicode_FromUnicode(\\r\\n            |                                          ^~~~~~~~~~~~~~~~~~~~~\\r\\n      error: command '/usr/bin/x86_64-linux-gnu-gcc' failed with exit code 1\\r\\n      [end of output]\\r\\n  \\r\\n  note: This error originates from a subprocess, and is likely not a problem with pip.\\r\\nerror: legacy-install-failure\\r\\n\\r\\n× Encountered error while trying to install package.\\r\\n╰─> lightfm\\r\\n\\r\\nnote: This is an issue with the package mentioned above, not pip.\\r\\nhint: See above for output from the failure.\\r\\n\\r\\n\\r\\n</details>\\r\\n\",\n",
       "  'reactions': {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50180/reactions',\n",
       "   'total_count': 0,\n",
       "   '+1': 0,\n",
       "   '-1': 0,\n",
       "   'laugh': 0,\n",
       "   'hooray': 0,\n",
       "   'confused': 0,\n",
       "   'heart': 0,\n",
       "   'rocket': 0,\n",
       "   'eyes': 0},\n",
       "  'timeline_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50180/timeline',\n",
       "  'performed_via_github_app': None,\n",
       "  'state_reason': None},\n",
       " {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50179',\n",
       "  'repository_url': 'https://api.github.com/repos/pandas-dev/pandas',\n",
       "  'labels_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50179/labels{/name}',\n",
       "  'comments_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50179/comments',\n",
       "  'events_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50179/events',\n",
       "  'html_url': 'https://github.com/pandas-dev/pandas/pull/50179',\n",
       "  'id': 1489502444,\n",
       "  'node_id': 'PR_kwDOAA0YD85FD1Ou',\n",
       "  'number': 50179,\n",
       "  'title': 'BUG: fixed issue where Index was being accessed with DatetimeIndex attribute',\n",
       "  'user': {'login': 'bilaljai',\n",
       "   'id': 55934929,\n",
       "   'node_id': 'MDQ6VXNlcjU1OTM0OTI5',\n",
       "   'avatar_url': 'https://avatars.githubusercontent.com/u/55934929?v=4',\n",
       "   'gravatar_id': '',\n",
       "   'url': 'https://api.github.com/users/bilaljai',\n",
       "   'html_url': 'https://github.com/bilaljai',\n",
       "   'followers_url': 'https://api.github.com/users/bilaljai/followers',\n",
       "   'following_url': 'https://api.github.com/users/bilaljai/following{/other_user}',\n",
       "   'gists_url': 'https://api.github.com/users/bilaljai/gists{/gist_id}',\n",
       "   'starred_url': 'https://api.github.com/users/bilaljai/starred{/owner}{/repo}',\n",
       "   'subscriptions_url': 'https://api.github.com/users/bilaljai/subscriptions',\n",
       "   'organizations_url': 'https://api.github.com/users/bilaljai/orgs',\n",
       "   'repos_url': 'https://api.github.com/users/bilaljai/repos',\n",
       "   'events_url': 'https://api.github.com/users/bilaljai/events{/privacy}',\n",
       "   'received_events_url': 'https://api.github.com/users/bilaljai/received_events',\n",
       "   'type': 'User',\n",
       "   'site_admin': False},\n",
       "  'labels': [],\n",
       "  'state': 'open',\n",
       "  'locked': False,\n",
       "  'assignee': None,\n",
       "  'assignees': [],\n",
       "  'milestone': None,\n",
       "  'comments': 0,\n",
       "  'created_at': '2022-12-11T05:56:03Z',\n",
       "  'updated_at': '2022-12-11T05:58:36Z',\n",
       "  'closed_at': None,\n",
       "  'author_association': 'NONE',\n",
       "  'active_lock_reason': None,\n",
       "  'draft': False,\n",
       "  'pull_request': {'url': 'https://api.github.com/repos/pandas-dev/pandas/pulls/50179',\n",
       "   'html_url': 'https://github.com/pandas-dev/pandas/pull/50179',\n",
       "   'diff_url': 'https://github.com/pandas-dev/pandas/pull/50179.diff',\n",
       "   'patch_url': 'https://github.com/pandas-dev/pandas/pull/50179.patch',\n",
       "   'merged_at': None},\n",
       "  'body': '- [ ] Closes #49925 \\r\\n- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature\\r\\n- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).\\r\\n- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.\\r\\n- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.\\r\\n\\r\\nChanges: updated the dates method in pandas/tseries/holiday.py to include a check to prevent the DatetimeIndex attribute\\r\\nfrom being accessed by an Index object.  The bug occurs if the holidays land outside the provided date range which results in an empty DatetimeIndex object being converted to an Index object. The updated dates method now checks if the object contains elements before allowing access to a DatetimeIndex attribute because such an object will be of type Index. Furthermore, the Index object is converted back into a DatetimeIndex object. A test was added to pandas/tests/tseries/holiday/test_holiday.py to ensure that the intended behavior is occurring. \\r\\n',\n",
       "  'reactions': {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50179/reactions',\n",
       "   'total_count': 0,\n",
       "   '+1': 0,\n",
       "   '-1': 0,\n",
       "   'laugh': 0,\n",
       "   'hooray': 0,\n",
       "   'confused': 0,\n",
       "   'heart': 0,\n",
       "   'rocket': 0,\n",
       "   'eyes': 0},\n",
       "  'timeline_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50179/timeline',\n",
       "  'performed_via_github_app': None,\n",
       "  'state_reason': None},\n",
       " {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50178',\n",
       "  'repository_url': 'https://api.github.com/repos/pandas-dev/pandas',\n",
       "  'labels_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50178/labels{/name}',\n",
       "  'comments_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50178/comments',\n",
       "  'events_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50178/events',\n",
       "  'html_url': 'https://github.com/pandas-dev/pandas/pull/50178',\n",
       "  'id': 1489125708,\n",
       "  'node_id': 'PR_kwDOAA0YD85FCcfG',\n",
       "  'number': 50178,\n",
       "  'title': 'DEPR: deprecate Index.is_integer',\n",
       "  'user': {'login': 'topper-123',\n",
       "   'id': 26364415,\n",
       "   'node_id': 'MDQ6VXNlcjI2MzY0NDE1',\n",
       "   'avatar_url': 'https://avatars.githubusercontent.com/u/26364415?v=4',\n",
       "   'gravatar_id': '',\n",
       "   'url': 'https://api.github.com/users/topper-123',\n",
       "   'html_url': 'https://github.com/topper-123',\n",
       "   'followers_url': 'https://api.github.com/users/topper-123/followers',\n",
       "   'following_url': 'https://api.github.com/users/topper-123/following{/other_user}',\n",
       "   'gists_url': 'https://api.github.com/users/topper-123/gists{/gist_id}',\n",
       "   'starred_url': 'https://api.github.com/users/topper-123/starred{/owner}{/repo}',\n",
       "   'subscriptions_url': 'https://api.github.com/users/topper-123/subscriptions',\n",
       "   'organizations_url': 'https://api.github.com/users/topper-123/orgs',\n",
       "   'repos_url': 'https://api.github.com/users/topper-123/repos',\n",
       "   'events_url': 'https://api.github.com/users/topper-123/events{/privacy}',\n",
       "   'received_events_url': 'https://api.github.com/users/topper-123/received_events',\n",
       "   'type': 'User',\n",
       "   'site_admin': False},\n",
       "  'labels': [],\n",
       "  'state': 'open',\n",
       "  'locked': False,\n",
       "  'assignee': None,\n",
       "  'assignees': [],\n",
       "  'milestone': None,\n",
       "  'comments': 0,\n",
       "  'created_at': '2022-12-10T23:37:41Z',\n",
       "  'updated_at': '2022-12-10T23:37:41Z',\n",
       "  'closed_at': None,\n",
       "  'author_association': 'CONTRIBUTOR',\n",
       "  'active_lock_reason': None,\n",
       "  'draft': False,\n",
       "  'pull_request': {'url': 'https://api.github.com/repos/pandas-dev/pandas/pulls/50178',\n",
       "   'html_url': 'https://github.com/pandas-dev/pandas/pull/50178',\n",
       "   'diff_url': 'https://github.com/pandas-dev/pandas/pull/50178.diff',\n",
       "   'patch_url': 'https://github.com/pandas-dev/pandas/pull/50178.patch',\n",
       "   'merged_at': None},\n",
       "  'body': '- [ ] progress towards #50042\\r\\n- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature\\r\\n- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).\\r\\n- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.\\r\\n- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.\\r\\n',\n",
       "  'reactions': {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50178/reactions',\n",
       "   'total_count': 0,\n",
       "   '+1': 0,\n",
       "   '-1': 0,\n",
       "   'laugh': 0,\n",
       "   'hooray': 0,\n",
       "   'confused': 0,\n",
       "   'heart': 0,\n",
       "   'rocket': 0,\n",
       "   'eyes': 0},\n",
       "  'timeline_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50178/timeline',\n",
       "  'performed_via_github_app': None,\n",
       "  'state_reason': None},\n",
       " {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50177',\n",
       "  'repository_url': 'https://api.github.com/repos/pandas-dev/pandas',\n",
       "  'labels_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50177/labels{/name}',\n",
       "  'comments_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50177/comments',\n",
       "  'events_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50177/events',\n",
       "  'html_url': 'https://github.com/pandas-dev/pandas/pull/50177',\n",
       "  'id': 1489123159,\n",
       "  'node_id': 'PR_kwDOAA0YD85FCb5H',\n",
       "  'number': 50177,\n",
       "  'title': 'BUG: ensure connection does not close before retrieving results from io.sql.execute.',\n",
       "  'user': {'login': 'cdcadman',\n",
       "   'id': 51368516,\n",
       "   'node_id': 'MDQ6VXNlcjUxMzY4NTE2',\n",
       "   'avatar_url': 'https://avatars.githubusercontent.com/u/51368516?v=4',\n",
       "   'gravatar_id': '',\n",
       "   'url': 'https://api.github.com/users/cdcadman',\n",
       "   'html_url': 'https://github.com/cdcadman',\n",
       "   'followers_url': 'https://api.github.com/users/cdcadman/followers',\n",
       "   'following_url': 'https://api.github.com/users/cdcadman/following{/other_user}',\n",
       "   'gists_url': 'https://api.github.com/users/cdcadman/gists{/gist_id}',\n",
       "   'starred_url': 'https://api.github.com/users/cdcadman/starred{/owner}{/repo}',\n",
       "   'subscriptions_url': 'https://api.github.com/users/cdcadman/subscriptions',\n",
       "   'organizations_url': 'https://api.github.com/users/cdcadman/orgs',\n",
       "   'repos_url': 'https://api.github.com/users/cdcadman/repos',\n",
       "   'events_url': 'https://api.github.com/users/cdcadman/events{/privacy}',\n",
       "   'received_events_url': 'https://api.github.com/users/cdcadman/received_events',\n",
       "   'type': 'User',\n",
       "   'site_admin': False},\n",
       "  'labels': [],\n",
       "  'state': 'open',\n",
       "  'locked': False,\n",
       "  'assignee': None,\n",
       "  'assignees': [],\n",
       "  'milestone': None,\n",
       "  'comments': 0,\n",
       "  'created_at': '2022-12-10T23:35:11Z',\n",
       "  'updated_at': '2022-12-11T04:57:52Z',\n",
       "  'closed_at': None,\n",
       "  'author_association': 'CONTRIBUTOR',\n",
       "  'active_lock_reason': None,\n",
       "  'draft': False,\n",
       "  'pull_request': {'url': 'https://api.github.com/repos/pandas-dev/pandas/pulls/50177',\n",
       "   'html_url': 'https://github.com/pandas-dev/pandas/pull/50177',\n",
       "   'diff_url': 'https://github.com/pandas-dev/pandas/pull/50177.diff',\n",
       "   'patch_url': 'https://github.com/pandas-dev/pandas/pull/50177.patch',\n",
       "   'merged_at': None},\n",
       "  'body': \"- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)\\r\\n- [X] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature\\r\\n- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).\\r\\n- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.\\r\\n- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.\\r\\n\\r\\nThis fixes a bug which I introduced in #49531, and it builds on an open PR #49967 which fixes another bug I introduced.  This PR's bug is in the method `pandas.io.sql.execute`, which is documented in the user guide.  I updated the documentation.  I had to increase the minimum sqlalchemy version to 1.4.45, because 1.4.44 has a bug which would cause `pandas.io.sql.execute` to fail after my change.  The sqlalchemy bug was fixed in https://github.com/sqlalchemy/sqlalchemy/issues/8963.\\r\\n\",\n",
       "  'reactions': {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50177/reactions',\n",
       "   'total_count': 0,\n",
       "   '+1': 0,\n",
       "   '-1': 0,\n",
       "   'laugh': 0,\n",
       "   'hooray': 0,\n",
       "   'confused': 0,\n",
       "   'heart': 0,\n",
       "   'rocket': 0,\n",
       "   'eyes': 0},\n",
       "  'timeline_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50177/timeline',\n",
       "  'performed_via_github_app': None,\n",
       "  'state_reason': None},\n",
       " {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50176',\n",
       "  'repository_url': 'https://api.github.com/repos/pandas-dev/pandas',\n",
       "  'labels_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50176/labels{/name}',\n",
       "  'comments_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50176/comments',\n",
       "  'events_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50176/events',\n",
       "  'html_url': 'https://github.com/pandas-dev/pandas/pull/50176',\n",
       "  'id': 1489118936,\n",
       "  'node_id': 'PR_kwDOAA0YD85FCa6D',\n",
       "  'number': 50176,\n",
       "  'title': 'DEPR: deprecate Index.is_boolean',\n",
       "  'user': {'login': 'topper-123',\n",
       "   'id': 26364415,\n",
       "   'node_id': 'MDQ6VXNlcjI2MzY0NDE1',\n",
       "   'avatar_url': 'https://avatars.githubusercontent.com/u/26364415?v=4',\n",
       "   'gravatar_id': '',\n",
       "   'url': 'https://api.github.com/users/topper-123',\n",
       "   'html_url': 'https://github.com/topper-123',\n",
       "   'followers_url': 'https://api.github.com/users/topper-123/followers',\n",
       "   'following_url': 'https://api.github.com/users/topper-123/following{/other_user}',\n",
       "   'gists_url': 'https://api.github.com/users/topper-123/gists{/gist_id}',\n",
       "   'starred_url': 'https://api.github.com/users/topper-123/starred{/owner}{/repo}',\n",
       "   'subscriptions_url': 'https://api.github.com/users/topper-123/subscriptions',\n",
       "   'organizations_url': 'https://api.github.com/users/topper-123/orgs',\n",
       "   'repos_url': 'https://api.github.com/users/topper-123/repos',\n",
       "   'events_url': 'https://api.github.com/users/topper-123/events{/privacy}',\n",
       "   'received_events_url': 'https://api.github.com/users/topper-123/received_events',\n",
       "   'type': 'User',\n",
       "   'site_admin': False},\n",
       "  'labels': [],\n",
       "  'state': 'open',\n",
       "  'locked': False,\n",
       "  'assignee': None,\n",
       "  'assignees': [],\n",
       "  'milestone': None,\n",
       "  'comments': 0,\n",
       "  'created_at': '2022-12-10T23:30:57Z',\n",
       "  'updated_at': '2022-12-10T23:30:57Z',\n",
       "  'closed_at': None,\n",
       "  'author_association': 'CONTRIBUTOR',\n",
       "  'active_lock_reason': None,\n",
       "  'draft': False,\n",
       "  'pull_request': {'url': 'https://api.github.com/repos/pandas-dev/pandas/pulls/50176',\n",
       "   'html_url': 'https://github.com/pandas-dev/pandas/pull/50176',\n",
       "   'diff_url': 'https://github.com/pandas-dev/pandas/pull/50176.diff',\n",
       "   'patch_url': 'https://github.com/pandas-dev/pandas/pull/50176.patch',\n",
       "   'merged_at': None},\n",
       "  'body': '- [ ] progress towards #50042\\r\\n- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature\\r\\n- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).\\r\\n- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.\\r\\n- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.\\r\\n',\n",
       "  'reactions': {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50176/reactions',\n",
       "   'total_count': 0,\n",
       "   '+1': 0,\n",
       "   '-1': 0,\n",
       "   'laugh': 0,\n",
       "   'hooray': 0,\n",
       "   'confused': 0,\n",
       "   'heart': 0,\n",
       "   'rocket': 0,\n",
       "   'eyes': 0},\n",
       "  'timeline_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50176/timeline',\n",
       "  'performed_via_github_app': None,\n",
       "  'state_reason': None},\n",
       " {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50175',\n",
       "  'repository_url': 'https://api.github.com/repos/pandas-dev/pandas',\n",
       "  'labels_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50175/labels{/name}',\n",
       "  'comments_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50175/comments',\n",
       "  'events_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50175/events',\n",
       "  'html_url': 'https://github.com/pandas-dev/pandas/pull/50175',\n",
       "  'id': 1489104096,\n",
       "  'node_id': 'PR_kwDOAA0YD85FCXbf',\n",
       "  'number': 50175,\n",
       "  'title': 'BUG: isetitem coercing df columns to object',\n",
       "  'user': {'login': 'phofl',\n",
       "   'id': 61934744,\n",
       "   'node_id': 'MDQ6VXNlcjYxOTM0NzQ0',\n",
       "   'avatar_url': 'https://avatars.githubusercontent.com/u/61934744?v=4',\n",
       "   'gravatar_id': '',\n",
       "   'url': 'https://api.github.com/users/phofl',\n",
       "   'html_url': 'https://github.com/phofl',\n",
       "   'followers_url': 'https://api.github.com/users/phofl/followers',\n",
       "   'following_url': 'https://api.github.com/users/phofl/following{/other_user}',\n",
       "   'gists_url': 'https://api.github.com/users/phofl/gists{/gist_id}',\n",
       "   'starred_url': 'https://api.github.com/users/phofl/starred{/owner}{/repo}',\n",
       "   'subscriptions_url': 'https://api.github.com/users/phofl/subscriptions',\n",
       "   'organizations_url': 'https://api.github.com/users/phofl/orgs',\n",
       "   'repos_url': 'https://api.github.com/users/phofl/repos',\n",
       "   'events_url': 'https://api.github.com/users/phofl/events{/privacy}',\n",
       "   'received_events_url': 'https://api.github.com/users/phofl/received_events',\n",
       "   'type': 'User',\n",
       "   'site_admin': False},\n",
       "  'labels': [{'id': 2822098,\n",
       "    'node_id': 'MDU6TGFiZWwyODIyMDk4',\n",
       "    'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Indexing',\n",
       "    'name': 'Indexing',\n",
       "    'color': '0b02e1',\n",
       "    'default': False,\n",
       "    'description': 'Related to indexing on series/frames, not to indexes themselves'},\n",
       "   {'id': 1049312478,\n",
       "    'node_id': 'MDU6TGFiZWwxMDQ5MzEyNDc4',\n",
       "    'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/DataFrame',\n",
       "    'name': 'DataFrame',\n",
       "    'color': '370f77',\n",
       "    'default': False,\n",
       "    'description': 'DataFrame data structure'},\n",
       "   {'id': 1817503692,\n",
       "    'node_id': 'MDU6TGFiZWwxODE3NTAzNjky',\n",
       "    'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/NA%20-%20MaskedArrays',\n",
       "    'name': 'NA - MaskedArrays',\n",
       "    'color': '8cc645',\n",
       "    'default': False,\n",
       "    'description': 'Related to pd.NA and nullable extension arrays'}],\n",
       "  'state': 'open',\n",
       "  'locked': False,\n",
       "  'assignee': None,\n",
       "  'assignees': [],\n",
       "  'milestone': None,\n",
       "  'comments': 0,\n",
       "  'created_at': '2022-12-10T23:16:44Z',\n",
       "  'updated_at': '2022-12-10T23:17:00Z',\n",
       "  'closed_at': None,\n",
       "  'author_association': 'MEMBER',\n",
       "  'active_lock_reason': None,\n",
       "  'draft': False,\n",
       "  'pull_request': {'url': 'https://api.github.com/repos/pandas-dev/pandas/pulls/50175',\n",
       "   'html_url': 'https://github.com/pandas-dev/pandas/pull/50175',\n",
       "   'diff_url': 'https://github.com/pandas-dev/pandas/pull/50175.diff',\n",
       "   'patch_url': 'https://github.com/pandas-dev/pandas/pull/50175.patch',\n",
       "   'merged_at': None},\n",
       "  'body': '- [x] closes #49922 (Replace xxxx with the GitHub issue number)\\r\\n- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature\\r\\n- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).\\r\\n- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.\\r\\n- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.\\r\\n',\n",
       "  'reactions': {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50175/reactions',\n",
       "   'total_count': 0,\n",
       "   '+1': 0,\n",
       "   '-1': 0,\n",
       "   'laugh': 0,\n",
       "   'hooray': 0,\n",
       "   'confused': 0,\n",
       "   'heart': 0,\n",
       "   'rocket': 0,\n",
       "   'eyes': 0},\n",
       "  'timeline_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50175/timeline',\n",
       "  'performed_via_github_app': None,\n",
       "  'state_reason': None},\n",
       " {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50174',\n",
       "  'repository_url': 'https://api.github.com/repos/pandas-dev/pandas',\n",
       "  'labels_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50174/labels{/name}',\n",
       "  'comments_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50174/comments',\n",
       "  'events_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50174/events',\n",
       "  'html_url': 'https://github.com/pandas-dev/pandas/pull/50174',\n",
       "  'id': 1489000364,\n",
       "  'node_id': 'PR_kwDOAA0YD85FB_Ps',\n",
       "  'number': 50174,\n",
       "  'title': 'Fixed issue with writing Decimal type to excel file - Issue #49598',\n",
       "  'user': {'login': 'alexanderwertman3',\n",
       "   'id': 56045351,\n",
       "   'node_id': 'MDQ6VXNlcjU2MDQ1MzUx',\n",
       "   'avatar_url': 'https://avatars.githubusercontent.com/u/56045351?v=4',\n",
       "   'gravatar_id': '',\n",
       "   'url': 'https://api.github.com/users/alexanderwertman3',\n",
       "   'html_url': 'https://github.com/alexanderwertman3',\n",
       "   'followers_url': 'https://api.github.com/users/alexanderwertman3/followers',\n",
       "   'following_url': 'https://api.github.com/users/alexanderwertman3/following{/other_user}',\n",
       "   'gists_url': 'https://api.github.com/users/alexanderwertman3/gists{/gist_id}',\n",
       "   'starred_url': 'https://api.github.com/users/alexanderwertman3/starred{/owner}{/repo}',\n",
       "   'subscriptions_url': 'https://api.github.com/users/alexanderwertman3/subscriptions',\n",
       "   'organizations_url': 'https://api.github.com/users/alexanderwertman3/orgs',\n",
       "   'repos_url': 'https://api.github.com/users/alexanderwertman3/repos',\n",
       "   'events_url': 'https://api.github.com/users/alexanderwertman3/events{/privacy}',\n",
       "   'received_events_url': 'https://api.github.com/users/alexanderwertman3/received_events',\n",
       "   'type': 'User',\n",
       "   'site_admin': False},\n",
       "  'labels': [],\n",
       "  'state': 'open',\n",
       "  'locked': False,\n",
       "  'assignee': None,\n",
       "  'assignees': [],\n",
       "  'milestone': None,\n",
       "  'comments': 0,\n",
       "  'created_at': '2022-12-10T21:40:02Z',\n",
       "  'updated_at': '2022-12-10T21:40:02Z',\n",
       "  'closed_at': None,\n",
       "  'author_association': 'NONE',\n",
       "  'active_lock_reason': None,\n",
       "  'draft': False,\n",
       "  'pull_request': {'url': 'https://api.github.com/repos/pandas-dev/pandas/pulls/50174',\n",
       "   'html_url': 'https://github.com/pandas-dev/pandas/pull/50174',\n",
       "   'diff_url': 'https://github.com/pandas-dev/pandas/pull/50174.diff',\n",
       "   'patch_url': 'https://github.com/pandas-dev/pandas/pull/50174.patch',\n",
       "   'merged_at': None},\n",
       "  'body': '- [ ] closes #49598  (Replace xxxx with the GitHub issue number)\\r\\n- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature\\r\\n- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).\\r\\n- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.\\r\\n- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.\\r\\n',\n",
       "  'reactions': {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50174/reactions',\n",
       "   'total_count': 0,\n",
       "   '+1': 0,\n",
       "   '-1': 0,\n",
       "   'laugh': 0,\n",
       "   'hooray': 0,\n",
       "   'confused': 0,\n",
       "   'heart': 0,\n",
       "   'rocket': 0,\n",
       "   'eyes': 0},\n",
       "  'timeline_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50174/timeline',\n",
       "  'performed_via_github_app': None,\n",
       "  'state_reason': None},\n",
       " {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50173',\n",
       "  'repository_url': 'https://api.github.com/repos/pandas-dev/pandas',\n",
       "  'labels_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50173/labels{/name}',\n",
       "  'comments_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50173/comments',\n",
       "  'events_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50173/events',\n",
       "  'html_url': 'https://github.com/pandas-dev/pandas/pull/50173',\n",
       "  'id': 1488974843,\n",
       "  'node_id': 'PR_kwDOAA0YD85FB5S1',\n",
       "  'number': 50173,\n",
       "  'title': 'BUG: df.setitem raising for ea bool df indexer',\n",
       "  'user': {'login': 'phofl',\n",
       "   'id': 61934744,\n",
       "   'node_id': 'MDQ6VXNlcjYxOTM0NzQ0',\n",
       "   'avatar_url': 'https://avatars.githubusercontent.com/u/61934744?v=4',\n",
       "   'gravatar_id': '',\n",
       "   'url': 'https://api.github.com/users/phofl',\n",
       "   'html_url': 'https://github.com/phofl',\n",
       "   'followers_url': 'https://api.github.com/users/phofl/followers',\n",
       "   'following_url': 'https://api.github.com/users/phofl/following{/other_user}',\n",
       "   'gists_url': 'https://api.github.com/users/phofl/gists{/gist_id}',\n",
       "   'starred_url': 'https://api.github.com/users/phofl/starred{/owner}{/repo}',\n",
       "   'subscriptions_url': 'https://api.github.com/users/phofl/subscriptions',\n",
       "   'organizations_url': 'https://api.github.com/users/phofl/orgs',\n",
       "   'repos_url': 'https://api.github.com/users/phofl/repos',\n",
       "   'events_url': 'https://api.github.com/users/phofl/events{/privacy}',\n",
       "   'received_events_url': 'https://api.github.com/users/phofl/received_events',\n",
       "   'type': 'User',\n",
       "   'site_admin': False},\n",
       "  'labels': [{'id': 2822098,\n",
       "    'node_id': 'MDU6TGFiZWwyODIyMDk4',\n",
       "    'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Indexing',\n",
       "    'name': 'Indexing',\n",
       "    'color': '0b02e1',\n",
       "    'default': False,\n",
       "    'description': 'Related to indexing on series/frames, not to indexes themselves'},\n",
       "   {'id': 1049312478,\n",
       "    'node_id': 'MDU6TGFiZWwxMDQ5MzEyNDc4',\n",
       "    'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/DataFrame',\n",
       "    'name': 'DataFrame',\n",
       "    'color': '370f77',\n",
       "    'default': False,\n",
       "    'description': 'DataFrame data structure'},\n",
       "   {'id': 1817503692,\n",
       "    'node_id': 'MDU6TGFiZWwxODE3NTAzNjky',\n",
       "    'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/NA%20-%20MaskedArrays',\n",
       "    'name': 'NA - MaskedArrays',\n",
       "    'color': '8cc645',\n",
       "    'default': False,\n",
       "    'description': 'Related to pd.NA and nullable extension arrays'}],\n",
       "  'state': 'open',\n",
       "  'locked': False,\n",
       "  'assignee': None,\n",
       "  'assignees': [],\n",
       "  'milestone': None,\n",
       "  'comments': 0,\n",
       "  'created_at': '2022-12-10T21:16:12Z',\n",
       "  'updated_at': '2022-12-10T21:16:12Z',\n",
       "  'closed_at': None,\n",
       "  'author_association': 'MEMBER',\n",
       "  'active_lock_reason': None,\n",
       "  'draft': False,\n",
       "  'pull_request': {'url': 'https://api.github.com/repos/pandas-dev/pandas/pulls/50173',\n",
       "   'html_url': 'https://github.com/pandas-dev/pandas/pull/50173',\n",
       "   'diff_url': 'https://github.com/pandas-dev/pandas/pull/50173.diff',\n",
       "   'patch_url': 'https://github.com/pandas-dev/pandas/pull/50173.patch',\n",
       "   'merged_at': None},\n",
       "  'body': '- [x] closes #47125 (Replace xxxx with the GitHub issue number)\\r\\n- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature\\r\\n- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).\\r\\n- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.\\r\\n- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.\\r\\n',\n",
       "  'reactions': {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50173/reactions',\n",
       "   'total_count': 0,\n",
       "   '+1': 0,\n",
       "   '-1': 0,\n",
       "   'laugh': 0,\n",
       "   'hooray': 0,\n",
       "   'confused': 0,\n",
       "   'heart': 0,\n",
       "   'rocket': 0,\n",
       "   'eyes': 0},\n",
       "  'timeline_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50173/timeline',\n",
       "  'performed_via_github_app': None,\n",
       "  'state_reason': None},\n",
       " {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50171',\n",
       "  'repository_url': 'https://api.github.com/repos/pandas-dev/pandas',\n",
       "  'labels_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50171/labels{/name}',\n",
       "  'comments_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50171/comments',\n",
       "  'events_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50171/events',\n",
       "  'html_url': 'https://github.com/pandas-dev/pandas/pull/50171',\n",
       "  'id': 1488953048,\n",
       "  'node_id': 'PR_kwDOAA0YD85FB0MZ',\n",
       "  'number': 50171,\n",
       "  'title': 'BUG: iloc raising for ea series',\n",
       "  'user': {'login': 'phofl',\n",
       "   'id': 61934744,\n",
       "   'node_id': 'MDQ6VXNlcjYxOTM0NzQ0',\n",
       "   'avatar_url': 'https://avatars.githubusercontent.com/u/61934744?v=4',\n",
       "   'gravatar_id': '',\n",
       "   'url': 'https://api.github.com/users/phofl',\n",
       "   'html_url': 'https://github.com/phofl',\n",
       "   'followers_url': 'https://api.github.com/users/phofl/followers',\n",
       "   'following_url': 'https://api.github.com/users/phofl/following{/other_user}',\n",
       "   'gists_url': 'https://api.github.com/users/phofl/gists{/gist_id}',\n",
       "   'starred_url': 'https://api.github.com/users/phofl/starred{/owner}{/repo}',\n",
       "   'subscriptions_url': 'https://api.github.com/users/phofl/subscriptions',\n",
       "   'organizations_url': 'https://api.github.com/users/phofl/orgs',\n",
       "   'repos_url': 'https://api.github.com/users/phofl/repos',\n",
       "   'events_url': 'https://api.github.com/users/phofl/events{/privacy}',\n",
       "   'received_events_url': 'https://api.github.com/users/phofl/received_events',\n",
       "   'type': 'User',\n",
       "   'site_admin': False},\n",
       "  'labels': [{'id': 2822098,\n",
       "    'node_id': 'MDU6TGFiZWwyODIyMDk4',\n",
       "    'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Indexing',\n",
       "    'name': 'Indexing',\n",
       "    'color': '0b02e1',\n",
       "    'default': False,\n",
       "    'description': 'Related to indexing on series/frames, not to indexes themselves'},\n",
       "   {'id': 1817503692,\n",
       "    'node_id': 'MDU6TGFiZWwxODE3NTAzNjky',\n",
       "    'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/NA%20-%20MaskedArrays',\n",
       "    'name': 'NA - MaskedArrays',\n",
       "    'color': '8cc645',\n",
       "    'default': False,\n",
       "    'description': 'Related to pd.NA and nullable extension arrays'}],\n",
       "  'state': 'open',\n",
       "  'locked': False,\n",
       "  'assignee': None,\n",
       "  'assignees': [],\n",
       "  'milestone': None,\n",
       "  'comments': 0,\n",
       "  'created_at': '2022-12-10T20:55:32Z',\n",
       "  'updated_at': '2022-12-11T10:04:43Z',\n",
       "  'closed_at': None,\n",
       "  'author_association': 'MEMBER',\n",
       "  'active_lock_reason': None,\n",
       "  'draft': False,\n",
       "  'pull_request': {'url': 'https://api.github.com/repos/pandas-dev/pandas/pulls/50171',\n",
       "   'html_url': 'https://github.com/pandas-dev/pandas/pull/50171',\n",
       "   'diff_url': 'https://github.com/pandas-dev/pandas/pull/50171.diff',\n",
       "   'patch_url': 'https://github.com/pandas-dev/pandas/pull/50171.patch',\n",
       "   'merged_at': None},\n",
       "  'body': '- [x] closes #49521 (Replace xxxx with the GitHub issue number)\\r\\n- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature\\r\\n- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).\\r\\n- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.\\r\\n- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.\\r\\n',\n",
       "  'reactions': {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50171/reactions',\n",
       "   'total_count': 0,\n",
       "   '+1': 0,\n",
       "   '-1': 0,\n",
       "   'laugh': 0,\n",
       "   'hooray': 0,\n",
       "   'confused': 0,\n",
       "   'heart': 0,\n",
       "   'rocket': 0,\n",
       "   'eyes': 0},\n",
       "  'timeline_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50171/timeline',\n",
       "  'performed_via_github_app': None,\n",
       "  'state_reason': None},\n",
       " {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50170',\n",
       "  'repository_url': 'https://api.github.com/repos/pandas-dev/pandas',\n",
       "  'labels_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50170/labels{/name}',\n",
       "  'comments_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50170/comments',\n",
       "  'events_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50170/events',\n",
       "  'html_url': 'https://github.com/pandas-dev/pandas/pull/50170',\n",
       "  'id': 1488944244,\n",
       "  'node_id': 'PR_kwDOAA0YD85FByIq',\n",
       "  'number': 50170,\n",
       "  'title': 'BUG: loc for ea index raising with oob slice end',\n",
       "  'user': {'login': 'phofl',\n",
       "   'id': 61934744,\n",
       "   'node_id': 'MDQ6VXNlcjYxOTM0NzQ0',\n",
       "   'avatar_url': 'https://avatars.githubusercontent.com/u/61934744?v=4',\n",
       "   'gravatar_id': '',\n",
       "   'url': 'https://api.github.com/users/phofl',\n",
       "   'html_url': 'https://github.com/phofl',\n",
       "   'followers_url': 'https://api.github.com/users/phofl/followers',\n",
       "   'following_url': 'https://api.github.com/users/phofl/following{/other_user}',\n",
       "   'gists_url': 'https://api.github.com/users/phofl/gists{/gist_id}',\n",
       "   'starred_url': 'https://api.github.com/users/phofl/starred{/owner}{/repo}',\n",
       "   'subscriptions_url': 'https://api.github.com/users/phofl/subscriptions',\n",
       "   'organizations_url': 'https://api.github.com/users/phofl/orgs',\n",
       "   'repos_url': 'https://api.github.com/users/phofl/repos',\n",
       "   'events_url': 'https://api.github.com/users/phofl/events{/privacy}',\n",
       "   'received_events_url': 'https://api.github.com/users/phofl/received_events',\n",
       "   'type': 'User',\n",
       "   'site_admin': False},\n",
       "  'labels': [{'id': 2822098,\n",
       "    'node_id': 'MDU6TGFiZWwyODIyMDk4',\n",
       "    'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Indexing',\n",
       "    'name': 'Indexing',\n",
       "    'color': '0b02e1',\n",
       "    'default': False,\n",
       "    'description': 'Related to indexing on series/frames, not to indexes themselves'},\n",
       "   {'id': 1218227310,\n",
       "    'node_id': 'MDU6TGFiZWwxMjE4MjI3MzEw',\n",
       "    'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Index',\n",
       "    'name': 'Index',\n",
       "    'color': 'e99695',\n",
       "    'default': False,\n",
       "    'description': 'Related to the Index class or subclasses'},\n",
       "   {'id': 1817503692,\n",
       "    'node_id': 'MDU6TGFiZWwxODE3NTAzNjky',\n",
       "    'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/NA%20-%20MaskedArrays',\n",
       "    'name': 'NA - MaskedArrays',\n",
       "    'color': '8cc645',\n",
       "    'default': False,\n",
       "    'description': 'Related to pd.NA and nullable extension arrays'}],\n",
       "  'state': 'open',\n",
       "  'locked': False,\n",
       "  'assignee': None,\n",
       "  'assignees': [],\n",
       "  'milestone': None,\n",
       "  'comments': 0,\n",
       "  'created_at': '2022-12-10T20:47:19Z',\n",
       "  'updated_at': '2022-12-10T20:47:28Z',\n",
       "  'closed_at': None,\n",
       "  'author_association': 'MEMBER',\n",
       "  'active_lock_reason': None,\n",
       "  'draft': False,\n",
       "  'pull_request': {'url': 'https://api.github.com/repos/pandas-dev/pandas/pulls/50170',\n",
       "   'html_url': 'https://github.com/pandas-dev/pandas/pull/50170',\n",
       "   'diff_url': 'https://github.com/pandas-dev/pandas/pull/50170.diff',\n",
       "   'patch_url': 'https://github.com/pandas-dev/pandas/pull/50170.patch',\n",
       "   'merged_at': None},\n",
       "  'body': '- [x] closes #50161 (Replace xxxx with the GitHub issue number)\\r\\n- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature\\r\\n- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).\\r\\n- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.\\r\\n- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.\\r\\n',\n",
       "  'reactions': {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50170/reactions',\n",
       "   'total_count': 0,\n",
       "   '+1': 0,\n",
       "   '-1': 0,\n",
       "   'laugh': 0,\n",
       "   'hooray': 0,\n",
       "   'confused': 0,\n",
       "   'heart': 0,\n",
       "   'rocket': 0,\n",
       "   'eyes': 0},\n",
       "  'timeline_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50170/timeline',\n",
       "  'performed_via_github_app': None,\n",
       "  'state_reason': None},\n",
       " {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50169',\n",
       "  'repository_url': 'https://api.github.com/repos/pandas-dev/pandas',\n",
       "  'labels_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50169/labels{/name}',\n",
       "  'comments_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50169/comments',\n",
       "  'events_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50169/events',\n",
       "  'html_url': 'https://github.com/pandas-dev/pandas/pull/50169',\n",
       "  'id': 1488901896,\n",
       "  'node_id': 'PR_kwDOAA0YD85FBoP5',\n",
       "  'number': 50169,\n",
       "  'title': 'Document generating compile_commands.json for C extensions',\n",
       "  'user': {'login': 'WillAyd',\n",
       "   'id': 609873,\n",
       "   'node_id': 'MDQ6VXNlcjYwOTg3Mw==',\n",
       "   'avatar_url': 'https://avatars.githubusercontent.com/u/609873?v=4',\n",
       "   'gravatar_id': '',\n",
       "   'url': 'https://api.github.com/users/WillAyd',\n",
       "   'html_url': 'https://github.com/WillAyd',\n",
       "   'followers_url': 'https://api.github.com/users/WillAyd/followers',\n",
       "   'following_url': 'https://api.github.com/users/WillAyd/following{/other_user}',\n",
       "   'gists_url': 'https://api.github.com/users/WillAyd/gists{/gist_id}',\n",
       "   'starred_url': 'https://api.github.com/users/WillAyd/starred{/owner}{/repo}',\n",
       "   'subscriptions_url': 'https://api.github.com/users/WillAyd/subscriptions',\n",
       "   'organizations_url': 'https://api.github.com/users/WillAyd/orgs',\n",
       "   'repos_url': 'https://api.github.com/users/WillAyd/repos',\n",
       "   'events_url': 'https://api.github.com/users/WillAyd/events{/privacy}',\n",
       "   'received_events_url': 'https://api.github.com/users/WillAyd/received_events',\n",
       "   'type': 'User',\n",
       "   'site_admin': False},\n",
       "  'labels': [],\n",
       "  'state': 'open',\n",
       "  'locked': False,\n",
       "  'assignee': None,\n",
       "  'assignees': [],\n",
       "  'milestone': None,\n",
       "  'comments': 0,\n",
       "  'created_at': '2022-12-10T20:06:40Z',\n",
       "  'updated_at': '2022-12-10T20:06:40Z',\n",
       "  'closed_at': None,\n",
       "  'author_association': 'MEMBER',\n",
       "  'active_lock_reason': None,\n",
       "  'draft': False,\n",
       "  'pull_request': {'url': 'https://api.github.com/repos/pandas-dev/pandas/pulls/50169',\n",
       "   'html_url': 'https://github.com/pandas-dev/pandas/pull/50169',\n",
       "   'diff_url': 'https://github.com/pandas-dev/pandas/pull/50169.diff',\n",
       "   'patch_url': 'https://github.com/pandas-dev/pandas/pull/50169.patch',\n",
       "   'merged_at': None},\n",
       "  'body': None,\n",
       "  'reactions': {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50169/reactions',\n",
       "   'total_count': 0,\n",
       "   '+1': 0,\n",
       "   '-1': 0,\n",
       "   'laugh': 0,\n",
       "   'hooray': 0,\n",
       "   'confused': 0,\n",
       "   'heart': 0,\n",
       "   'rocket': 0,\n",
       "   'eyes': 0},\n",
       "  'timeline_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50169/timeline',\n",
       "  'performed_via_github_app': None,\n",
       "  'state_reason': None},\n",
       " {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50166',\n",
       "  'repository_url': 'https://api.github.com/repos/pandas-dev/pandas',\n",
       "  'labels_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50166/labels{/name}',\n",
       "  'comments_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50166/comments',\n",
       "  'events_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50166/events',\n",
       "  'html_url': 'https://github.com/pandas-dev/pandas/issues/50166',\n",
       "  'id': 1488515984,\n",
       "  'node_id': 'I_kwDOAA0YD85YuPOQ',\n",
       "  'number': 50166,\n",
       "  'title': \"BUG: reindex __cinit__() got multiple values for keyword argument 'verify_integrity'\",\n",
       "  'user': {'login': 'Trackingerror',\n",
       "   'id': 103377869,\n",
       "   'node_id': 'U_kgDOBilrzQ',\n",
       "   'avatar_url': 'https://avatars.githubusercontent.com/u/103377869?v=4',\n",
       "   'gravatar_id': '',\n",
       "   'url': 'https://api.github.com/users/Trackingerror',\n",
       "   'html_url': 'https://github.com/Trackingerror',\n",
       "   'followers_url': 'https://api.github.com/users/Trackingerror/followers',\n",
       "   'following_url': 'https://api.github.com/users/Trackingerror/following{/other_user}',\n",
       "   'gists_url': 'https://api.github.com/users/Trackingerror/gists{/gist_id}',\n",
       "   'starred_url': 'https://api.github.com/users/Trackingerror/starred{/owner}{/repo}',\n",
       "   'subscriptions_url': 'https://api.github.com/users/Trackingerror/subscriptions',\n",
       "   'organizations_url': 'https://api.github.com/users/Trackingerror/orgs',\n",
       "   'repos_url': 'https://api.github.com/users/Trackingerror/repos',\n",
       "   'events_url': 'https://api.github.com/users/Trackingerror/events{/privacy}',\n",
       "   'received_events_url': 'https://api.github.com/users/Trackingerror/received_events',\n",
       "   'type': 'User',\n",
       "   'site_admin': False},\n",
       "  'labels': [{'id': 76811,\n",
       "    'node_id': 'MDU6TGFiZWw3NjgxMQ==',\n",
       "    'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Bug',\n",
       "    'name': 'Bug',\n",
       "    'color': 'e10c02',\n",
       "    'default': False,\n",
       "    'description': None},\n",
       "   {'id': 307649777,\n",
       "    'node_id': 'MDU6TGFiZWwzMDc2NDk3Nzc=',\n",
       "    'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Needs%20Info',\n",
       "    'name': 'Needs Info',\n",
       "    'color': '207de5',\n",
       "    'default': False,\n",
       "    'description': 'Clarification about behavior needed to assess issue'}],\n",
       "  'state': 'open',\n",
       "  'locked': False,\n",
       "  'assignee': None,\n",
       "  'assignees': [],\n",
       "  'milestone': None,\n",
       "  'comments': 1,\n",
       "  'created_at': '2022-12-10T14:05:17Z',\n",
       "  'updated_at': '2022-12-10T20:02:06Z',\n",
       "  'closed_at': None,\n",
       "  'author_association': 'NONE',\n",
       "  'active_lock_reason': None,\n",
       "  'body': '### Pandas version checks\\n\\n- [X] I have checked that this issue has not already been reported.\\n\\n- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.\\n\\n- [X] I have confirmed this bug exists on the main branch of pandas.\\n\\n\\n### Reproducible Example\\n\\n```python\\ndef import_histo(data_histo):\\r\\n    dfh = pd.read_csv(\\'Datas/\\' + data_histo,sep = \";\", low_memory = False)\\r\\n    data_cols = [\\'var1\\', \\'var2\\', \\'var3\\', \\'var4\\', \\'var5\\', \\'var6\\', \\'var7\\', \\'var8\\', \\'var9\\']\\r\\n    dfh.reindex(data_cols, axis=\\'columns\\')\\r\\n    dfh.set_index(\\'var1\\',inplace = True)\\r\\n    return dfh\\r\\n\\r\\nfor file_var in file_list:\\r\\n    df2 = import_histo(file_var)\\r\\n    df = pd.concat([df,df2])\\n```\\n\\n\\n### Issue Description\\n\\n---------------------------------------------------------------------------\\r\\nTypeError                                 Traceback (most recent call last)\\r\\nCell In [1], line 41\\r\\n     39 #description:\\r\\n     40 for file_var in file_list:\\r\\n---> 41     df2 = import_histo(file_var)\\r\\n     42     df = pd.concat([df,df2])\\r\\n     44 df = df.sort_values(\\'var1\\', ascending = False)\\r\\n\\r\\nCell In [1], line 23, in import_histo(data_histo)\\r\\n     21 data_cols = [\\'var1\\', \\'var2\\', \\'var3\\', \\'var4\\', \\'var5\\', \\'var6\\', \\'var7\\', \\'var8\\', \\'var9\\']\\r\\n     22 #dfh.reindex(data_cols, axis=\\'columns\\')\\r\\n---> 23 dfh = dfh.reindex(columns = data_cols)\\r\\n     24 #dfh.reindex()\\r\\n     25 dfh.set_index(\\'annee_numero_de_tirage\\',inplace = True)\\r\\n\\r\\nFile /private/var/containers/Bundle/Application/9181B29D-3843-4BCA-B130-BECC651D8757/Carnets.app/Library/lib/python3.11/site-packages/pandas-1.5.2-py3.11-macosx-11.5-x86_64.egg/pandas/util/_decorators.py:347, in rewrite_axis_style_signature.<locals>.decorate.<locals>.wrapper(*args, **kwargs)\\r\\n    345 @wraps(func)\\r\\n    346 def wrapper(*args, **kwargs) -> Callable[..., Any]:\\r\\n--> 347     return func(*args, **kwargs)\\r\\n\\r\\nFile /private/var/containers/Bundle/Application/9181B29D-3843-4BCA-B130-BECC651D8757/Carnets.app/Library/lib/python3.11/site-packages/pandas-1.5.2-py3.11-macosx-11.5-x86_64.egg/pandas/core/frame.py:5202, in DataFrame.reindex(self, *args, **kwargs)\\r\\n   5200 kwargs.pop(\"axis\", None)\\r\\n   5201 kwargs.pop(\"labels\", None)\\r\\n-> 5202 return super().reindex(**kwargs)\\r\\n\\r\\nFile /private/var/containers/Bundle/Application/9181B29D-3843-4BCA-B130-BECC651D8757/Carnets.app/Library/lib/python3.11/site-packages/pandas-1.5.2-py3.11-macosx-11.5-x86_64.egg/pandas/core/generic.py:5289, in NDFrame.reindex(self, *args, **kwargs)\\r\\n   5286     return self._reindex_multi(axes, copy, fill_value)\\r\\n   5288 # perform the reindex on the axes\\r\\n-> 5289 return self._reindex_axes(\\r\\n   5290     axes, level, limit, tolerance, method, fill_value, copy\\r\\n   5291 ).__finalize__(self, method=\"reindex\")\\r\\n\\r\\nFile /private/var/containers/Bundle/Application/9181B29D-3843-4BCA-B130-BECC651D8757/Carnets.app/Library/lib/python3.11/site-packages/pandas-1.5.2-py3.11-macosx-11.5-x86_64.egg/pandas/core/frame.py:4995, in DataFrame._reindex_axes(self, axes, level, limit, tolerance, method, fill_value, copy)\\r\\n   4993 columns = axes[\"columns\"]\\r\\n   4994 if columns is not None:\\r\\n-> 4995     frame = frame._reindex_columns(\\r\\n   4996         columns, method, copy, level, fill_value, limit, tolerance\\r\\n   4997     )\\r\\n   4999 index = axes[\"index\"]\\r\\n   5000 if index is not None:\\r\\n\\r\\nFile /private/var/containers/Bundle/Application/9181B29D-3843-4BCA-B130-BECC651D8757/Carnets.app/Library/lib/python3.11/site-packages/pandas-1.5.2-py3.11-macosx-11.5-x86_64.egg/pandas/core/frame.py:5040, in DataFrame._reindex_columns(self, new_columns, method, copy, level, fill_value, limit, tolerance)\\r\\n   5027 def _reindex_columns(\\r\\n   5028     self,\\r\\n   5029     new_columns,\\r\\n   (...)\\r\\n   5035     tolerance=None,\\r\\n   5036 ):\\r\\n   5037     new_columns, indexer = self.columns.reindex(\\r\\n   5038         new_columns, method=method, level=level, limit=limit, tolerance=tolerance\\r\\n   5039     )\\r\\n-> 5040     return self._reindex_with_indexers(\\r\\n   5041         {1: [new_columns, indexer]},\\r\\n   5042         copy=copy,\\r\\n   5043         fill_value=fill_value,\\r\\n   5044         allow_dups=False,\\r\\n   5045     )\\r\\n\\r\\nFile /private/var/containers/Bundle/Application/9181B29D-3843-4BCA-B130-BECC651D8757/Carnets.app/Library/lib/python3.11/site-packages/pandas-1.5.2-py3.11-macosx-11.5-x86_64.egg/pandas/core/generic.py:5355, in NDFrame._reindex_with_indexers(self, reindexers, fill_value, copy, allow_dups)\\r\\n   5352     indexer = ensure_platform_int(indexer)\\r\\n   5354 # TODO: speed up on homogeneous DataFrame objects (see _reindex_multi)\\r\\n-> 5355 new_data = new_data.reindex_indexer(\\r\\n   5356     index,\\r\\n   5357     indexer,\\r\\n   5358     axis=baxis,\\r\\n   5359     fill_value=fill_value,\\r\\n   5360     allow_dups=allow_dups,\\r\\n   5361     copy=copy,\\r\\n   5362 )\\r\\n   5363 # If we\\'ve made a copy once, no need to make another one\\r\\n   5364 copy = False\\r\\n\\r\\nFile /private/var/containers/Bundle/Application/9181B29D-3843-4BCA-B130-BECC651D8757/Carnets.app/Library/lib/python3.11/site-packages/pandas-1.5.2-py3.11-macosx-11.5-x86_64.egg/pandas/core/internals/managers.py:769, in BaseBlockManager.reindex_indexer(self, new_axis, indexer, axis, fill_value, allow_dups, copy, only_slice, use_na_proxy)\\r\\n    766 new_axes = list(self.axes)\\r\\n    767 new_axes[axis] = new_axis\\r\\n--> 769 new_mgr = type(self).from_blocks(new_blocks, new_axes, new_refs, parent=parent)\\r\\n    770 if axis == 1:\\r\\n    771     # We can avoid the need to rebuild these\\r\\n    772     new_mgr._blknos = self.blknos.copy()\\r\\n\\r\\nFile /private/var/containers/Bundle/Application/9181B29D-3843-4BCA-B130-BECC651D8757/Carnets.app/Library/lib/python3.11/site-packages/pandas-1.5.2-py3.11-macosx-11.5-x86_64.egg/pandas/core/internals/managers.py:1075, in BlockManager.from_blocks(cls, blocks, axes, refs, parent)\\r\\n   1071 \"\"\"\\r\\n   1072 Constructor for BlockManager and SingleBlockManager with same signature.\\r\\n   1073 \"\"\"\\r\\n   1074 parent = parent if _using_copy_on_write() else None\\r\\n-> 1075 return cls(blocks, axes, refs, parent, verify_integrity=False)\\r\\n\\r\\nFile /private/var/containers/Bundle/Application/9181B29D-3843-4BCA-B130-BECC651D8757/Carnets.app/Library/lib/python3.11/site-packages/pandas-1.5.2-py3.11-macosx-11.5-x86_64.egg/pandas/_libs/internals.pyx:680, in pandas._libs.internals.BlockManager.__cinit__()\\r\\n\\r\\nTypeError: __cinit__() got multiple values for keyword argument \\'verify_integrity\\'\\n\\n### Expected Behavior\\n\\nThis code used to perfectly works on 1.3.X Pandas.\\r\\nIt returns a Dataframe reindexed by columns.\\r\\nI updated the version on my tablet device and on my laptop and I now have this bug.\\r\\nI\\'ve found nothing on the web regarding this issue.\\n\\n### Installed Versions\\n\\n<details>\\r\\n\\r\\nINSTALLED VERSIONS\\r\\n------------------\\r\\ncommit           : 8dab54d6573f7186ff0c3b6364d5e4dd635ff3e7\\r\\npython           : 3.11.0.final.0\\r\\npython-bits      : 64\\r\\nOS               : Darwin\\r\\nOS-release       : 22.1.0\\r\\nVersion          : Darwin Kernel Version 22.1.0: Thu Oct  6 19:33:23 PDT 2022; root:xnu-8792.42.7~1/RELEASE_ARM64_T8103\\r\\nmachine          : iPad13,16\\r\\nprocessor        : arm\\r\\nbyteorder        : little\\r\\nLC_ALL           : UTF-8\\r\\nLANG             : None\\r\\nLOCALE           : None.UTF-8\\r\\n\\r\\npandas           : 1.5.2\\r\\nnumpy            : 1.24.0.dev0+1021.g61098c4a4\\r\\npytz             : 2022.6\\r\\ndateutil         : 2.8.2\\r\\nsetuptools       : 65.6.3\\r\\npip              : 22.3.1\\r\\nCython           : 0.29.32\\r\\npytest           : None\\r\\nhypothesis       : None\\r\\nsphinx           : None\\r\\nblosc            : None\\r\\nfeather          : None\\r\\nxlsxwriter       : None\\r\\nlxml.etree       : 4.9.1\\r\\nhtml5lib         : 1.1\\r\\npymysql          : None\\r\\npsycopg2         : None\\r\\njinja2           : 3.1.2\\r\\nIPython          : 8.5.0\\r\\npandas_datareader: None\\r\\nbs4              : 4.11.1\\r\\nbottleneck       : None\\r\\nbrotli           : None\\r\\nfastparquet      : None\\r\\nfsspec           : None\\r\\ngcsfs            : None\\r\\nmatplotlib       : 3.7.0.dev641+g768ff8450\\r\\nnumba            : None\\r\\nnumexpr          : None\\r\\nodfpy            : None\\r\\nopenpyxl         : 3.0.9\\r\\npandas_gbq       : None\\r\\npyarrow          : None\\r\\npyreadstat       : None\\r\\npyxlsb           : None\\r\\ns3fs             : None\\r\\nscipy            : None\\r\\nsnappy           : None\\r\\nsqlalchemy       : None\\r\\ntables           : None\\r\\ntabulate         : None\\r\\nxarray           : None\\r\\nxlrd             : None\\r\\nxlwt             : None\\r\\nzstandard        : None \\r\\ntzdata : None \\r\\n\\r\\n</details>\\r\\n',\n",
       "  'reactions': {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50166/reactions',\n",
       "   'total_count': 0,\n",
       "   '+1': 0,\n",
       "   '-1': 0,\n",
       "   'laugh': 0,\n",
       "   'hooray': 0,\n",
       "   'confused': 0,\n",
       "   'heart': 0,\n",
       "   'rocket': 0,\n",
       "   'eyes': 0},\n",
       "  'timeline_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50166/timeline',\n",
       "  'performed_via_github_app': None,\n",
       "  'state_reason': None},\n",
       " {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50165',\n",
       "  'repository_url': 'https://api.github.com/repos/pandas-dev/pandas',\n",
       "  'labels_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50165/labels{/name}',\n",
       "  'comments_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50165/comments',\n",
       "  'events_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50165/events',\n",
       "  'html_url': 'https://github.com/pandas-dev/pandas/issues/50165',\n",
       "  'id': 1488366831,\n",
       "  'node_id': 'I_kwDOAA0YD85Ytqzv',\n",
       "  'number': 50165,\n",
       "  'title': 'BUG: loc raising error for MultiIndex with bool indexer _and first level of object dtype_ ',\n",
       "  'user': {'login': 'toobaz',\n",
       "   'id': 1224492,\n",
       "   'node_id': 'MDQ6VXNlcjEyMjQ0OTI=',\n",
       "   'avatar_url': 'https://avatars.githubusercontent.com/u/1224492?v=4',\n",
       "   'gravatar_id': '',\n",
       "   'url': 'https://api.github.com/users/toobaz',\n",
       "   'html_url': 'https://github.com/toobaz',\n",
       "   'followers_url': 'https://api.github.com/users/toobaz/followers',\n",
       "   'following_url': 'https://api.github.com/users/toobaz/following{/other_user}',\n",
       "   'gists_url': 'https://api.github.com/users/toobaz/gists{/gist_id}',\n",
       "   'starred_url': 'https://api.github.com/users/toobaz/starred{/owner}{/repo}',\n",
       "   'subscriptions_url': 'https://api.github.com/users/toobaz/subscriptions',\n",
       "   'organizations_url': 'https://api.github.com/users/toobaz/orgs',\n",
       "   'repos_url': 'https://api.github.com/users/toobaz/repos',\n",
       "   'events_url': 'https://api.github.com/users/toobaz/events{/privacy}',\n",
       "   'received_events_url': 'https://api.github.com/users/toobaz/received_events',\n",
       "   'type': 'User',\n",
       "   'site_admin': False},\n",
       "  'labels': [{'id': 76811,\n",
       "    'node_id': 'MDU6TGFiZWw3NjgxMQ==',\n",
       "    'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Bug',\n",
       "    'name': 'Bug',\n",
       "    'color': 'e10c02',\n",
       "    'default': False,\n",
       "    'description': None},\n",
       "   {'id': 1954720290,\n",
       "    'node_id': 'MDU6TGFiZWwxOTU0NzIwMjkw',\n",
       "    'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Needs%20Triage',\n",
       "    'name': 'Needs Triage',\n",
       "    'color': '0052cc',\n",
       "    'default': False,\n",
       "    'description': 'Issue that has not been reviewed by a pandas team member'}],\n",
       "  'state': 'open',\n",
       "  'locked': False,\n",
       "  'assignee': None,\n",
       "  'assignees': [],\n",
       "  'milestone': None,\n",
       "  'comments': 0,\n",
       "  'created_at': '2022-12-10T10:41:57Z',\n",
       "  'updated_at': '2022-12-10T10:43:06Z',\n",
       "  'closed_at': None,\n",
       "  'author_association': 'MEMBER',\n",
       "  'active_lock_reason': None,\n",
       "  'body': '### Pandas version checks\\r\\n\\r\\n- [X] I have checked that this issue has not already been reported.\\r\\n\\r\\n- [ ] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.\\r\\n\\r\\n- [X] I have confirmed this bug exists on the main branch of pandas.\\r\\n\\r\\n\\r\\n### Reproducible Example\\r\\n\\r\\n```python\\r\\nvalues = {\\r\\n          False :\\r\\n          {\\'A\\' : range(2), \\'B\\' : range(2)},\\r\\n          True :\\r\\n          {\\'A\\' : range(2), \\'B\\' : range(2)}\\r\\n         }\\r\\n\\r\\nvalues = pd.concat({k : pd.DataFrame(values[k]) for k in values})\\r\\n\\r\\nvalues.loc[False]\\r\\n```\\r\\n\\r\\n```\\r\\n---------------------------------------------------------------------------\\r\\nKeyError                                  Traceback (most recent call last)\\r\\n<ipython-input-40-cca3b469b9a2> in <module>\\r\\n----> 1 values.loc[False]\\r\\n\\r\\n~/.local/lib/python3.9/site-packages/pandas/core/indexing.py in __getitem__(self, key)\\r\\n    965 \\r\\n    966             maybe_callable = com.apply_if_callable(key, self.obj)\\r\\n--> 967             return self._getitem_axis(maybe_callable, axis=axis)\\r\\n    968 \\r\\n    969     def _is_scalar_access(self, key: tuple):\\r\\n\\r\\n~/.local/lib/python3.9/site-packages/pandas/core/indexing.py in _getitem_axis(self, key, axis)\\r\\n   1199 \\r\\n   1200         # fall thru to straight lookup\\r\\n-> 1201         self._validate_key(key, axis)\\r\\n   1202         return self._get_label(key, axis=axis)\\r\\n   1203 \\r\\n\\r\\n~/.local/lib/python3.9/site-packages/pandas/core/indexing.py in _validate_key(self, key, axis)\\r\\n   1009             or self.obj._get_axis(axis).dtype.name == \"boolean\"\\r\\n   1010         ):\\r\\n-> 1011             raise KeyError(\\r\\n   1012                 f\"{key}: boolean label can not be used without a boolean index\"\\r\\n   1013             )\\r\\n\\r\\nKeyError: \\'False: boolean label can not be used without a boolean index\\'\\r\\n\\r\\n```\\r\\n\\r\\n### Issue Description\\r\\n\\r\\nIssue https://github.com/pandas-dev/pandas/issues/47687 was fixed in https://github.com/pandas-dev/pandas/pull/49766 by checking that if the index is a MultiIndex its first level must be of type bool.\\r\\n\\r\\nHowever, this still seems inconsistent with other dtypes, and code paths - for instance ``values.loc[(False, ), :]`` works flawlessly with the example I am providing above.\\r\\n\\r\\nI argue that the original sin is in\\r\\n\\r\\nhttps://github.com/pandas-dev/pandas/pull/40726/files#diff-07282871a235456baecd3a55a43339b0da374ca270228b5530f5449aabdca2e1R968\\r\\n\\r\\nThat PR was meant to solve a specific issue, that is, that in some cases a boolean label is casted to int, and from there a series of unexpected behaviors occur.\\r\\n\\r\\nBut there is no reason on earth why Boolean labels shouldn\\'t just follow the rules for labels of other dtypes, we just want to ensure that any undesired casting is avoided across any code path.\\r\\n\\r\\nBy the way, I checked that neither of the two problematic examples in https://github.com/pandas-dev/pandas/issues/20432 work any more, _even removing all code from ``_LocIndexer._validate_key()``_.\\r\\n\\r\\nSo I think we should do it.\\r\\n\\r\\nThoughts @mroeschke ?\\r\\n\\r\\n### Expected Behavior\\r\\n\\r\\nSame as ``values.loc[(False, ), :]``\\r\\n\\r\\n### Installed Versions\\r\\n\\r\\n<details>\\r\\n\\r\\nINSTALLED VERSIONS\\r\\n------------------\\r\\ncommit           : cab77e3d6c122bc11e13dcbd700cbbfbdba31fa4\\r\\npython           : 3.9.2.final.0\\r\\npython-bits      : 64\\r\\nOS               : Linux\\r\\nOS-release       : 5.18.0-0.deb11.4-amd64\\r\\nVersion          : #1 SMP PREEMPT_DYNAMIC Debian 5.18.16-1~bpo11+1 (2022-08-12)\\r\\nmachine          : x86_64\\r\\nprocessor        : \\r\\nbyteorder        : little\\r\\nLC_ALL           : None\\r\\nLANG             : it_IT.UTF-8\\r\\nLOCALE           : it_IT.UTF-8\\r\\n\\r\\npandas           : 2.0.0.dev0+881.gcab77e3d6c.dirty\\r\\nnumpy            : 1.23.3\\r\\npytz             : 2021.1\\r\\ndateutil         : 2.8.2\\r\\nsetuptools       : 52.0.0\\r\\npip              : 20.3.4\\r\\nCython           : 0.29.32\\r\\npytest           : 6.0.2\\r\\nhypothesis       : 5.43.3\\r\\nsphinx           : 3.4.3\\r\\nblosc            : 1.9.2\\r\\nfeather          : None\\r\\nxlsxwriter       : 1.1.2\\r\\nlxml.etree       : 4.6.3\\r\\nhtml5lib         : 1.1\\r\\npymysql          : None\\r\\npsycopg2         : 2.8.6\\r\\njinja2           : 2.11.3\\r\\nIPython          : 7.20.0\\r\\npandas_datareader: None\\r\\nbs4              : 4.8.2\\r\\nbottleneck       : 1.3.4\\r\\nbrotli           : 1.0.9\\r\\nfastparquet      : None\\r\\nfsspec           : 0.8.4\\r\\ngcsfs            : None\\r\\nmatplotlib       : 3.3.4\\r\\nnumba            : None\\r\\nnumexpr          : 2.7.2\\r\\nodfpy            : None\\r\\nopenpyxl         : 3.0.3\\r\\npandas_gbq       : None\\r\\npyarrow          : None\\r\\npyreadstat       : 1.1.0\\r\\npyxlsb           : None\\r\\ns3fs             : None\\r\\nscipy            : 1.6.0\\r\\nsnappy           : None\\r\\nsqlalchemy       : 1.3.22\\r\\ntables           : 3.6.1\\r\\ntabulate         : None\\r\\nxarray           : 0.16.1\\r\\nxlrd             : 2.0.1\\r\\nzstandard        : None\\r\\ntzdata           : None\\r\\nqtpy             : 1.9.0\\r\\npyqt5            : None\\r\\n\\r\\n\\r\\n</details>\\r\\n',\n",
       "  'reactions': {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50165/reactions',\n",
       "   'total_count': 0,\n",
       "   '+1': 0,\n",
       "   '-1': 0,\n",
       "   'laugh': 0,\n",
       "   'hooray': 0,\n",
       "   'confused': 0,\n",
       "   'heart': 0,\n",
       "   'rocket': 0,\n",
       "   'eyes': 0},\n",
       "  'timeline_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50165/timeline',\n",
       "  'performed_via_github_app': None,\n",
       "  'state_reason': None},\n",
       " {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50164',\n",
       "  'repository_url': 'https://api.github.com/repos/pandas-dev/pandas',\n",
       "  'labels_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50164/labels{/name}',\n",
       "  'comments_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50164/comments',\n",
       "  'events_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50164/events',\n",
       "  'html_url': 'https://github.com/pandas-dev/pandas/pull/50164',\n",
       "  'id': 1488037847,\n",
       "  'node_id': 'PR_kwDOAA0YD85E-h6C',\n",
       "  'number': 50164,\n",
       "  'title': 'TYP #37715: Fix mypy errors in column.py',\n",
       "  'user': {'login': 'bang128',\n",
       "   'id': 71242233,\n",
       "   'node_id': 'MDQ6VXNlcjcxMjQyMjMz',\n",
       "   'avatar_url': 'https://avatars.githubusercontent.com/u/71242233?v=4',\n",
       "   'gravatar_id': '',\n",
       "   'url': 'https://api.github.com/users/bang128',\n",
       "   'html_url': 'https://github.com/bang128',\n",
       "   'followers_url': 'https://api.github.com/users/bang128/followers',\n",
       "   'following_url': 'https://api.github.com/users/bang128/following{/other_user}',\n",
       "   'gists_url': 'https://api.github.com/users/bang128/gists{/gist_id}',\n",
       "   'starred_url': 'https://api.github.com/users/bang128/starred{/owner}{/repo}',\n",
       "   'subscriptions_url': 'https://api.github.com/users/bang128/subscriptions',\n",
       "   'organizations_url': 'https://api.github.com/users/bang128/orgs',\n",
       "   'repos_url': 'https://api.github.com/users/bang128/repos',\n",
       "   'events_url': 'https://api.github.com/users/bang128/events{/privacy}',\n",
       "   'received_events_url': 'https://api.github.com/users/bang128/received_events',\n",
       "   'type': 'User',\n",
       "   'site_admin': False},\n",
       "  'labels': [],\n",
       "  'state': 'open',\n",
       "  'locked': False,\n",
       "  'assignee': None,\n",
       "  'assignees': [],\n",
       "  'milestone': None,\n",
       "  'comments': 1,\n",
       "  'created_at': '2022-12-10T05:57:38Z',\n",
       "  'updated_at': '2022-12-11T01:13:06Z',\n",
       "  'closed_at': None,\n",
       "  'author_association': 'CONTRIBUTOR',\n",
       "  'active_lock_reason': None,\n",
       "  'draft': False,\n",
       "  'pull_request': {'url': 'https://api.github.com/repos/pandas-dev/pandas/pulls/50164',\n",
       "   'html_url': 'https://github.com/pandas-dev/pandas/pull/50164',\n",
       "   'diff_url': 'https://github.com/pandas-dev/pandas/pull/50164.diff',\n",
       "   'patch_url': 'https://github.com/pandas-dev/pandas/pull/50164.patch',\n",
       "   'merged_at': None},\n",
       "  'body': '- [ ] xref #37715 \\r\\n- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature\\r\\n- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).\\r\\n- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.\\r\\n- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.\\r\\n',\n",
       "  'reactions': {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50164/reactions',\n",
       "   'total_count': 0,\n",
       "   '+1': 0,\n",
       "   '-1': 0,\n",
       "   'laugh': 0,\n",
       "   'hooray': 0,\n",
       "   'confused': 0,\n",
       "   'heart': 0,\n",
       "   'rocket': 0,\n",
       "   'eyes': 0},\n",
       "  'timeline_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50164/timeline',\n",
       "  'performed_via_github_app': None,\n",
       "  'state_reason': None},\n",
       " {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50163',\n",
       "  'repository_url': 'https://api.github.com/repos/pandas-dev/pandas',\n",
       "  'labels_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50163/labels{/name}',\n",
       "  'comments_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50163/comments',\n",
       "  'events_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50163/events',\n",
       "  'html_url': 'https://github.com/pandas-dev/pandas/issues/50163',\n",
       "  'id': 1487716798,\n",
       "  'node_id': 'I_kwDOAA0YD85YrMG-',\n",
       "  'number': 50163,\n",
       "  'title': 'BUG: Deprecation warning for bool/numeric concatenation seems too broadly scoped',\n",
       "  'user': {'login': 'vyasr',\n",
       "   'id': 1538165,\n",
       "   'node_id': 'MDQ6VXNlcjE1MzgxNjU=',\n",
       "   'avatar_url': 'https://avatars.githubusercontent.com/u/1538165?v=4',\n",
       "   'gravatar_id': '',\n",
       "   'url': 'https://api.github.com/users/vyasr',\n",
       "   'html_url': 'https://github.com/vyasr',\n",
       "   'followers_url': 'https://api.github.com/users/vyasr/followers',\n",
       "   'following_url': 'https://api.github.com/users/vyasr/following{/other_user}',\n",
       "   'gists_url': 'https://api.github.com/users/vyasr/gists{/gist_id}',\n",
       "   'starred_url': 'https://api.github.com/users/vyasr/starred{/owner}{/repo}',\n",
       "   'subscriptions_url': 'https://api.github.com/users/vyasr/subscriptions',\n",
       "   'organizations_url': 'https://api.github.com/users/vyasr/orgs',\n",
       "   'repos_url': 'https://api.github.com/users/vyasr/repos',\n",
       "   'events_url': 'https://api.github.com/users/vyasr/events{/privacy}',\n",
       "   'received_events_url': 'https://api.github.com/users/vyasr/received_events',\n",
       "   'type': 'User',\n",
       "   'site_admin': False},\n",
       "  'labels': [{'id': 76811,\n",
       "    'node_id': 'MDU6TGFiZWw3NjgxMQ==',\n",
       "    'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Bug',\n",
       "    'name': 'Bug',\n",
       "    'color': 'e10c02',\n",
       "    'default': False,\n",
       "    'description': None},\n",
       "   {'id': 1954720290,\n",
       "    'node_id': 'MDU6TGFiZWwxOTU0NzIwMjkw',\n",
       "    'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Needs%20Triage',\n",
       "    'name': 'Needs Triage',\n",
       "    'color': '0052cc',\n",
       "    'default': False,\n",
       "    'description': 'Issue that has not been reviewed by a pandas team member'}],\n",
       "  'state': 'open',\n",
       "  'locked': False,\n",
       "  'assignee': None,\n",
       "  'assignees': [],\n",
       "  'milestone': None,\n",
       "  'comments': 0,\n",
       "  'created_at': '2022-12-10T00:50:18Z',\n",
       "  'updated_at': '2022-12-10T01:00:06Z',\n",
       "  'closed_at': None,\n",
       "  'author_association': 'NONE',\n",
       "  'active_lock_reason': None,\n",
       "  'body': '### Pandas version checks\\r\\n\\r\\n- [X] I have checked that this issue has not already been reported.\\r\\n\\r\\n- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.\\r\\n\\r\\n- [ ] I have confirmed this bug exists on the main branch of pandas.\\r\\n\\r\\n\\r\\n### Reproducible Example\\r\\n\\r\\n```python\\r\\n>>> import pandas as pd\\r\\n>>> bool_df = pd.DataFrame({\"a\": [False]})\\r\\n>>> _ = pd.concat([bool_df, pd.DataFrame({\"a\": [1]})])\\r\\n>>> _ = pd.concat([bool_df, pd.DataFrame({\"a\": [1.0]})])\\r\\n>>> _ = pd.concat([bool_df, pd.DataFrame({\"a\": []}, dtype=\"int64\")])\\r\\n>>> _ = pd.concat([bool_df, pd.DataFrame({\"a\": []}, dtype=\"float64\")])\\r\\n<stdin>:1: FutureWarning: Behavior when concatenating bool-dtype and numeric-dtype arrays is deprecated; in a future version these will cast to object dtype (instead of coercing bools to numeric values). To retain the old behavior, explicitly cast bool-dtype arrays to numeric dtype.\\r\\n```\\r\\n\\r\\n\\r\\n### Issue Description\\r\\n\\r\\nI\\'m seeing a generic warning about how concatenating bool and numeric dtypes will stop casting to a numeric type, but it only seems to occur in very specific cases of inputs (specifically, for an empty float dtype with the bool data). I\\'m not sure if this is a bug in where a FutureWarning is being thrown or just a request for a more descriptive message with the warning, so apologies for perhaps misunderstanding the intent here. I looked through #39817 and #45101, and they seem to indicate an awareness of the difference between empty and non-empty columns, but I don\\'t know if that was meant to propagate to the deprecation or not.\\r\\n\\r\\n### Expected Behavior\\r\\n\\r\\nI would expect the same warning to be shown for int-bool or float-bool operations, and to be raised regardless of whether the object is empty, but based on my reading of the related issues I\\'m not sure if that expectation is valid. If the warnings is in fact being shown in the only relevant case, I would expect the message to more clearly indicate exactly what case is being deprecated (empty array, float data, etc).\\r\\n\\r\\n### Installed Versions\\r\\n\\r\\n<details>\\r\\n\\r\\nINSTALLED VERSIONS     \\r\\n------------------    \\r\\ncommit           : 8dab54d6573f7186ff0c3b6364d5e4dd635ff3e7\\r\\npython           : 3.8.15.final.0\\r\\npython-bits      : 64\\r\\nOS               : Linux\\r\\nOS-release       : 5.14.0-1054-oem\\r\\nVersion          : #61-Ubuntu SMP Fri Oct 14 13:05:50 UTC 2022\\r\\nmachine          : x86_64\\r\\nprocessor        : x86_64\\r\\nbyteorder        : little\\r\\nLC_ALL           : None\\r\\nLANG             : None\\r\\nLOCALE           : en_US.UTF-8\\r\\n\\r\\npandas           : 1.5.2\\r\\nnumpy            : 1.23.5\\r\\npytz             : 2022.6\\r\\ndateutil         : 2.8.2\\r\\nsetuptools       : 65.5.1\\r\\npip              : 22.3.1\\r\\nCython           : 0.29.32\\r\\npytest           : 7.2.0\\r\\nhypothesis       : 6.60.0\\r\\nsphinx           : 5.3.0\\r\\nblosc            : None\\r\\nfeather          : None\\r\\nxlsxwriter       : None\\r\\nlxml.etree       : 4.9.1\\r\\nhtml5lib         : None\\r\\npymysql          : None\\r\\npsycopg2         : None\\r\\njinja2           : 3.1.2\\r\\nIPython          : 8.7.0\\r\\npandas_datareader: None\\r\\nbs4              : 4.11.1\\r\\nbottleneck       : None\\r\\nbrotli           : \\r\\nfastparquet      : None\\r\\nfsspec           : 2022.11.0\\r\\ngcsfs            : None\\r\\nmatplotlib       : None\\r\\nnumba            : 0.56.4\\r\\nnumexpr          : None\\r\\nodfpy            : None\\r\\nopenpyxl         : None\\r\\npandas_gbq       : None\\r\\npyarrow          : 9.0.0\\r\\npyreadstat       : None\\r\\npyxlsb           : None\\r\\ns3fs             : 2022.11.0\\r\\nscipy            : 1.9.3\\r\\nsnappy           : \\r\\nsqlalchemy       : 1.4.44\\r\\ntables           : None\\r\\ntabulate         : 0.9.0\\r\\nxarray           : None\\r\\nxlrd             : None\\r\\nxlwt             : None\\r\\nzstandard        : None\\r\\ntzdata           : None\\r\\n\\r\\n\\r\\n</details>\\r\\n',\n",
       "  'reactions': {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50163/reactions',\n",
       "   'total_count': 0,\n",
       "   '+1': 0,\n",
       "   '-1': 0,\n",
       "   'laugh': 0,\n",
       "   'hooray': 0,\n",
       "   'confused': 0,\n",
       "   'heart': 0,\n",
       "   'rocket': 0,\n",
       "   'eyes': 0},\n",
       "  'timeline_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50163/timeline',\n",
       "  'performed_via_github_app': None,\n",
       "  'state_reason': None},\n",
       " {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50162',\n",
       "  'repository_url': 'https://api.github.com/repos/pandas-dev/pandas',\n",
       "  'labels_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50162/labels{/name}',\n",
       "  'comments_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50162/comments',\n",
       "  'events_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50162/events',\n",
       "  'html_url': 'https://github.com/pandas-dev/pandas/issues/50162',\n",
       "  'id': 1487612599,\n",
       "  'node_id': 'I_kwDOAA0YD85Yqyq3',\n",
       "  'number': 50162,\n",
       "  'title': 'BUG: In master, math with periodindex + index of offset produces wrong result',\n",
       "  'user': {'login': 'Dr-Irv',\n",
       "   'id': 15113894,\n",
       "   'node_id': 'MDQ6VXNlcjE1MTEzODk0',\n",
       "   'avatar_url': 'https://avatars.githubusercontent.com/u/15113894?v=4',\n",
       "   'gravatar_id': '',\n",
       "   'url': 'https://api.github.com/users/Dr-Irv',\n",
       "   'html_url': 'https://github.com/Dr-Irv',\n",
       "   'followers_url': 'https://api.github.com/users/Dr-Irv/followers',\n",
       "   'following_url': 'https://api.github.com/users/Dr-Irv/following{/other_user}',\n",
       "   'gists_url': 'https://api.github.com/users/Dr-Irv/gists{/gist_id}',\n",
       "   'starred_url': 'https://api.github.com/users/Dr-Irv/starred{/owner}{/repo}',\n",
       "   'subscriptions_url': 'https://api.github.com/users/Dr-Irv/subscriptions',\n",
       "   'organizations_url': 'https://api.github.com/users/Dr-Irv/orgs',\n",
       "   'repos_url': 'https://api.github.com/users/Dr-Irv/repos',\n",
       "   'events_url': 'https://api.github.com/users/Dr-Irv/events{/privacy}',\n",
       "   'received_events_url': 'https://api.github.com/users/Dr-Irv/received_events',\n",
       "   'type': 'User',\n",
       "   'site_admin': False},\n",
       "  'labels': [{'id': 76811,\n",
       "    'node_id': 'MDU6TGFiZWw3NjgxMQ==',\n",
       "    'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Bug',\n",
       "    'name': 'Bug',\n",
       "    'color': 'e10c02',\n",
       "    'default': False,\n",
       "    'description': None},\n",
       "   {'id': 60635328,\n",
       "    'node_id': 'MDU6TGFiZWw2MDYzNTMyOA==',\n",
       "    'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Period',\n",
       "    'name': 'Period',\n",
       "    'color': 'eb6420',\n",
       "    'default': False,\n",
       "    'description': 'Period data type'},\n",
       "   {'id': 1218227310,\n",
       "    'node_id': 'MDU6TGFiZWwxMjE4MjI3MzEw',\n",
       "    'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Index',\n",
       "    'name': 'Index',\n",
       "    'color': 'e99695',\n",
       "    'default': False,\n",
       "    'description': 'Related to the Index class or subclasses'},\n",
       "   {'id': 1954720290,\n",
       "    'node_id': 'MDU6TGFiZWwxOTU0NzIwMjkw',\n",
       "    'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Needs%20Triage',\n",
       "    'name': 'Needs Triage',\n",
       "    'color': '0052cc',\n",
       "    'default': False,\n",
       "    'description': 'Issue that has not been reviewed by a pandas team member'}],\n",
       "  'state': 'open',\n",
       "  'locked': False,\n",
       "  'assignee': None,\n",
       "  'assignees': [],\n",
       "  'milestone': None,\n",
       "  'comments': 0,\n",
       "  'created_at': '2022-12-09T23:20:48Z',\n",
       "  'updated_at': '2022-12-09T23:20:48Z',\n",
       "  'closed_at': None,\n",
       "  'author_association': 'CONTRIBUTOR',\n",
       "  'active_lock_reason': None,\n",
       "  'body': '### Pandas version checks\\n\\n- [X] I have checked that this issue has not already been reported.\\n\\n- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.\\n\\n- [X] I have confirmed this bug exists on the main branch of pandas.\\n\\n\\n### Reproducible Example\\n\\n```python\\nimport pandas as pd\\r\\np = pd.Period(\"2012-1-1\", freq=\"D\")\\r\\nas_period_index = pd.period_range(\"2012-1-1\", periods=10, freq=\"D\")\\r\\noffset_index = p - as_period_index\\r\\nprint(offset_index)\\r\\nresult = p + offset_index\\r\\nprint(result)\\n```\\n\\n\\n### Issue Description\\n\\nIn pandas 1.5.2, here is the result:\\r\\n```text\\r\\nIndex([ <0 * Days>,  <-1 * Day>, <-2 * Days>, <-3 * Days>, <-4 * Days>,\\r\\n       <-5 * Days>, <-6 * Days>, <-7 * Days>, <-8 * Days>, <-9 * Days>],\\r\\n      dtype=\\'object\\')\\r\\nPeriodIndex([\\'2012-01-01\\', \\'2011-12-31\\', \\'2011-12-30\\', \\'2011-12-29\\',\\r\\n             \\'2011-12-28\\', \\'2011-12-27\\', \\'2011-12-26\\', \\'2011-12-25\\',\\r\\n             \\'2011-12-24\\', \\'2011-12-23\\'],\\r\\n            dtype=\\'period[D]\\')\\r\\n```\\r\\nIn master version `2.0.0.dev0+881.gcab77e3d6c`\\r\\n```text\\r\\nIndex([ <0 * Days>,  <-1 * Day>, <-2 * Days>, <-3 * Days>, <-4 * Days>,\\r\\n       <-5 * Days>, <-6 * Days>, <-7 * Days>, <-8 * Days>, <-9 * Days>],\\r\\n      dtype=\\'object\\')\\r\\nIndex([2012-01-01, 2011-12-31, 2011-12-30, 2011-12-29, 2011-12-28, 2011-12-27,\\r\\n       2011-12-26, 2011-12-25, 2011-12-24, 2011-12-23],\\r\\n      dtype=\\'object\\')\\r\\n```\\r\\n\\r\\nSo the result used to be `PeriodIndex`, now it is `Index`.  This was picked up in our tests for `pandas-stubs`\\n\\n### Expected Behavior\\n\\nShould return `PeriodIndex` when you add offsets\\n\\n### Installed Versions\\n\\n<details>\\r\\n\\r\\nINSTALLED VERSIONS\\r\\n------------------\\r\\ncommit           : cab77e3d6c122bc11e13dcbd700cbbfbdba31fa4\\r\\npython           : 3.8.15.final.0\\r\\npython-bits      : 64\\r\\nOS               : Windows\\r\\nOS-release       : 10\\r\\nVersion          : 10.0.19045\\r\\nmachine          : AMD64\\r\\nprocessor        : Intel64 Family 6 Model 158 Stepping 13, GenuineIntel\\r\\nbyteorder        : little\\r\\nLC_ALL           : None\\r\\nLANG             : None\\r\\nLOCALE           : English_United States.1252\\r\\n\\r\\npandas           : 2.0.0.dev0+881.gcab77e3d6c\\r\\nnumpy            : 1.23.5\\r\\npytz             : 2022.6\\r\\ndateutil         : 2.8.2\\r\\nsetuptools       : 65.4.1\\r\\npip              : 22.3.1\\r\\nCython           : 0.29.32\\r\\npytest           : 7.2.0\\r\\nhypothesis       : 6.60.0\\r\\nsphinx           : 4.5.0\\r\\nblosc            : None\\r\\nfeather          : None\\r\\nxlsxwriter       : 3.0.3\\r\\nlxml.etree       : 4.9.1\\r\\nhtml5lib         : 1.1\\r\\npymysql          : 1.0.2\\r\\npsycopg2         : 2.9.3\\r\\njinja2           : 3.0.3\\r\\nIPython          : 8.7.0\\r\\npandas_datareader: 0.10.0\\r\\nbs4              : 4.11.1\\r\\nbottleneck       : 1.3.5\\r\\nbrotli           :\\r\\nfastparquet      : 2022.12.0\\r\\nfsspec           : 2021.11.0\\r\\ngcsfs            : 2021.11.0\\r\\nmatplotlib       : 3.6.2\\r\\nnumba            : 0.56.4\\r\\nnumexpr          : 2.8.0\\r\\nodfpy            : None\\r\\nopenpyxl         : 3.0.10\\r\\npandas_gbq       : 0.17.9\\r\\npyarrow          : 9.0.0\\r\\npyreadstat       : 1.2.0\\r\\npyxlsb           : 1.0.10\\r\\ns3fs             : 2021.11.0\\r\\nscipy            : 1.9.3\\r\\nsnappy           :\\r\\nsqlalchemy       : 1.4.44\\r\\ntables           : 3.7.0\\r\\ntabulate         : 0.9.0\\r\\nxarray           : 2022.12.0\\r\\nxlrd             : 2.0.1\\r\\nzstandard        : 0.19.0\\r\\ntzdata           : None\\r\\nqtpy             : None\\r\\npyqt5            : None\\r\\n</details>\\r\\n',\n",
       "  'reactions': {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50162/reactions',\n",
       "   'total_count': 0,\n",
       "   '+1': 0,\n",
       "   '-1': 0,\n",
       "   'laugh': 0,\n",
       "   'hooray': 0,\n",
       "   'confused': 0,\n",
       "   'heart': 0,\n",
       "   'rocket': 0,\n",
       "   'eyes': 0},\n",
       "  'timeline_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50162/timeline',\n",
       "  'performed_via_github_app': None,\n",
       "  'state_reason': None},\n",
       " {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50161',\n",
       "  'repository_url': 'https://api.github.com/repos/pandas-dev/pandas',\n",
       "  'labels_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50161/labels{/name}',\n",
       "  'comments_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50161/comments',\n",
       "  'events_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50161/events',\n",
       "  'html_url': 'https://github.com/pandas-dev/pandas/issues/50161',\n",
       "  'id': 1487594448,\n",
       "  'node_id': 'I_kwDOAA0YD85YquPQ',\n",
       "  'number': 50161,\n",
       "  'title': 'BUG: `Series` slicing can fail for extension array `Index`',\n",
       "  'user': {'login': 'jrbourbeau',\n",
       "   'id': 11656932,\n",
       "   'node_id': 'MDQ6VXNlcjExNjU2OTMy',\n",
       "   'avatar_url': 'https://avatars.githubusercontent.com/u/11656932?v=4',\n",
       "   'gravatar_id': '',\n",
       "   'url': 'https://api.github.com/users/jrbourbeau',\n",
       "   'html_url': 'https://github.com/jrbourbeau',\n",
       "   'followers_url': 'https://api.github.com/users/jrbourbeau/followers',\n",
       "   'following_url': 'https://api.github.com/users/jrbourbeau/following{/other_user}',\n",
       "   'gists_url': 'https://api.github.com/users/jrbourbeau/gists{/gist_id}',\n",
       "   'starred_url': 'https://api.github.com/users/jrbourbeau/starred{/owner}{/repo}',\n",
       "   'subscriptions_url': 'https://api.github.com/users/jrbourbeau/subscriptions',\n",
       "   'organizations_url': 'https://api.github.com/users/jrbourbeau/orgs',\n",
       "   'repos_url': 'https://api.github.com/users/jrbourbeau/repos',\n",
       "   'events_url': 'https://api.github.com/users/jrbourbeau/events{/privacy}',\n",
       "   'received_events_url': 'https://api.github.com/users/jrbourbeau/received_events',\n",
       "   'type': 'User',\n",
       "   'site_admin': False},\n",
       "  'labels': [{'id': 76811,\n",
       "    'node_id': 'MDU6TGFiZWw3NjgxMQ==',\n",
       "    'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Bug',\n",
       "    'name': 'Bug',\n",
       "    'color': 'e10c02',\n",
       "    'default': False,\n",
       "    'description': None},\n",
       "   {'id': 2822098,\n",
       "    'node_id': 'MDU6TGFiZWwyODIyMDk4',\n",
       "    'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Indexing',\n",
       "    'name': 'Indexing',\n",
       "    'color': '0b02e1',\n",
       "    'default': False,\n",
       "    'description': 'Related to indexing on series/frames, not to indexes themselves'},\n",
       "   {'id': 1218227310,\n",
       "    'node_id': 'MDU6TGFiZWwxMjE4MjI3MzEw',\n",
       "    'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Index',\n",
       "    'name': 'Index',\n",
       "    'color': 'e99695',\n",
       "    'default': False,\n",
       "    'description': 'Related to the Index class or subclasses'},\n",
       "   {'id': 1817503692,\n",
       "    'node_id': 'MDU6TGFiZWwxODE3NTAzNjky',\n",
       "    'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/NA%20-%20MaskedArrays',\n",
       "    'name': 'NA - MaskedArrays',\n",
       "    'color': '8cc645',\n",
       "    'default': False,\n",
       "    'description': 'Related to pd.NA and nullable extension arrays'}],\n",
       "  'state': 'open',\n",
       "  'locked': False,\n",
       "  'assignee': None,\n",
       "  'assignees': [],\n",
       "  'milestone': None,\n",
       "  'comments': 1,\n",
       "  'created_at': '2022-12-09T23:05:05Z',\n",
       "  'updated_at': '2022-12-10T20:36:47Z',\n",
       "  'closed_at': None,\n",
       "  'author_association': 'CONTRIBUTOR',\n",
       "  'active_lock_reason': None,\n",
       "  'body': '### Pandas version checks\\n\\n- [X] I have checked that this issue has not already been reported.\\n\\n- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.\\n\\n- [ ] I have confirmed this bug exists on the main branch of pandas.\\n\\n\\n### Reproducible Example\\n\\n```python\\nimport pandas as pd\\r\\n\\r\\ns = pd.Series(range(5), index=pd.Index(range(5), dtype=int))\\r\\nprint(s.loc[4:5])  # works\\r\\n\\r\\n# Fails with both `numpy`- and `pyarrow`-backed extension arrays for the index\\r\\ns = pd.Series(range(5), index=pd.Index(range(5), dtype=\"Int64\"))\\r\\nprint(s.loc[4:5])  # fails\\r\\n\\r\\ns = pd.Series(range(5), index=pd.Index(range(5), dtype=\"int64[pyarrow]\"))\\r\\nprint(s.loc[4:5])  # fails\\n```\\n\\n\\n### Issue Description\\n\\nI ran into an issue where slicing a series (via `.loc`) works for when using an index with `dtype=int`, but fails when I cast the index to use either `numpy`- or `pyarrow`-backed extension arrays. \\n\\n### Expected Behavior\\n\\nI think these are all valid `.loc` calls that should return the corresponding selection\\n\\n### Installed Versions\\n\\n<details>\\r\\n\\r\\n```\\r\\nINSTALLED VERSIONS\\r\\n------------------\\r\\ncommit           : 8dab54d6573f7186ff0c3b6364d5e4dd635ff3e7\\r\\npython           : 3.9.15.final.0\\r\\npython-bits      : 64\\r\\nOS               : Darwin\\r\\nOS-release       : 21.6.0\\r\\nVersion          : Darwin Kernel Version 21.6.0: Thu Sep 29 20:12:57 PDT 2022; root:xnu-8020.240.7~1/RELEASE_X86_64\\r\\nmachine          : x86_64\\r\\nprocessor        : i386\\r\\nbyteorder        : little\\r\\nLC_ALL           : None\\r\\nLANG             : en_US.UTF-8\\r\\nLOCALE           : en_US.UTF-8\\r\\n\\r\\npandas           : 1.5.2\\r\\nnumpy            : 1.23.5\\r\\npytz             : 2022.6\\r\\ndateutil         : 2.8.2\\r\\nsetuptools       : 59.8.0\\r\\npip              : 22.3.1\\r\\nCython           : None\\r\\npytest           : 7.2.0\\r\\nhypothesis       : None\\r\\nsphinx           : 4.5.0\\r\\nblosc            : None\\r\\nfeather          : None\\r\\nxlsxwriter       : None\\r\\nlxml.etree       : None\\r\\nhtml5lib         : 1.1\\r\\npymysql          : None\\r\\npsycopg2         : None\\r\\njinja2           : 3.1.2\\r\\nIPython          : 8.7.0\\r\\npandas_datareader: None\\r\\nbs4              : 4.11.1\\r\\nbottleneck       : None\\r\\nbrotli           :\\r\\nfastparquet      : 2022.11.0\\r\\nfsspec           : 2022.11.0\\r\\ngcsfs            : None\\r\\nmatplotlib       : None\\r\\nnumba            : None\\r\\nnumexpr          : 2.8.3\\r\\nodfpy            : None\\r\\nopenpyxl         : None\\r\\npandas_gbq       : None\\r\\npyarrow          : 10.0.1\\r\\npyreadstat       : None\\r\\npyxlsb           : None\\r\\ns3fs             : 2022.11.0\\r\\nscipy            : 1.9.3\\r\\nsnappy           :\\r\\nsqlalchemy       : 1.4.44\\r\\ntables           : 3.7.0\\r\\ntabulate         : None\\r\\nxarray           : 2022.11.0\\r\\nxlrd             : None\\r\\nxlwt             : None\\r\\nzstandard        : None\\r\\ntzdata           : None\\r\\n```\\r\\n</details>\\r\\n',\n",
       "  'reactions': {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50161/reactions',\n",
       "   'total_count': 0,\n",
       "   '+1': 0,\n",
       "   '-1': 0,\n",
       "   'laugh': 0,\n",
       "   'hooray': 0,\n",
       "   'confused': 0,\n",
       "   'heart': 0,\n",
       "   'rocket': 0,\n",
       "   'eyes': 0},\n",
       "  'timeline_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50161/timeline',\n",
       "  'performed_via_github_app': None,\n",
       "  'state_reason': None},\n",
       " {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50160',\n",
       "  'repository_url': 'https://api.github.com/repos/pandas-dev/pandas',\n",
       "  'labels_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50160/labels{/name}',\n",
       "  'comments_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50160/comments',\n",
       "  'events_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50160/events',\n",
       "  'html_url': 'https://github.com/pandas-dev/pandas/pull/50160',\n",
       "  'id': 1487068383,\n",
       "  'node_id': 'PR_kwDOAA0YD85E7EW0',\n",
       "  'number': 50160,\n",
       "  'title': 'Test ruff for linting',\n",
       "  'user': {'login': 'jorisvandenbossche',\n",
       "   'id': 1020496,\n",
       "   'node_id': 'MDQ6VXNlcjEwMjA0OTY=',\n",
       "   'avatar_url': 'https://avatars.githubusercontent.com/u/1020496?v=4',\n",
       "   'gravatar_id': '',\n",
       "   'url': 'https://api.github.com/users/jorisvandenbossche',\n",
       "   'html_url': 'https://github.com/jorisvandenbossche',\n",
       "   'followers_url': 'https://api.github.com/users/jorisvandenbossche/followers',\n",
       "   'following_url': 'https://api.github.com/users/jorisvandenbossche/following{/other_user}',\n",
       "   'gists_url': 'https://api.github.com/users/jorisvandenbossche/gists{/gist_id}',\n",
       "   'starred_url': 'https://api.github.com/users/jorisvandenbossche/starred{/owner}{/repo}',\n",
       "   'subscriptions_url': 'https://api.github.com/users/jorisvandenbossche/subscriptions',\n",
       "   'organizations_url': 'https://api.github.com/users/jorisvandenbossche/orgs',\n",
       "   'repos_url': 'https://api.github.com/users/jorisvandenbossche/repos',\n",
       "   'events_url': 'https://api.github.com/users/jorisvandenbossche/events{/privacy}',\n",
       "   'received_events_url': 'https://api.github.com/users/jorisvandenbossche/received_events',\n",
       "   'type': 'User',\n",
       "   'site_admin': False},\n",
       "  'labels': [],\n",
       "  'state': 'open',\n",
       "  'locked': False,\n",
       "  'assignee': None,\n",
       "  'assignees': [],\n",
       "  'milestone': None,\n",
       "  'comments': 0,\n",
       "  'created_at': '2022-12-09T16:49:22Z',\n",
       "  'updated_at': '2022-12-09T19:11:40Z',\n",
       "  'closed_at': None,\n",
       "  'author_association': 'MEMBER',\n",
       "  'active_lock_reason': None,\n",
       "  'draft': True,\n",
       "  'pull_request': {'url': 'https://api.github.com/repos/pandas-dev/pandas/pulls/50160',\n",
       "   'html_url': 'https://github.com/pandas-dev/pandas/pull/50160',\n",
       "   'diff_url': 'https://github.com/pandas-dev/pandas/pull/50160.diff',\n",
       "   'patch_url': 'https://github.com/pandas-dev/pandas/pull/50160.patch',\n",
       "   'merged_at': None},\n",
       "  'body': None,\n",
       "  'reactions': {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50160/reactions',\n",
       "   'total_count': 0,\n",
       "   '+1': 0,\n",
       "   '-1': 0,\n",
       "   'laugh': 0,\n",
       "   'hooray': 0,\n",
       "   'confused': 0,\n",
       "   'heart': 0,\n",
       "   'rocket': 0,\n",
       "   'eyes': 0},\n",
       "  'timeline_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50160/timeline',\n",
       "  'performed_via_github_app': None,\n",
       "  'state_reason': None},\n",
       " {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50158',\n",
       "  'repository_url': 'https://api.github.com/repos/pandas-dev/pandas',\n",
       "  'labels_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50158/labels{/name}',\n",
       "  'comments_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50158/comments',\n",
       "  'events_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50158/events',\n",
       "  'html_url': 'https://github.com/pandas-dev/pandas/pull/50158',\n",
       "  'id': 1487036953,\n",
       "  'node_id': 'PR_kwDOAA0YD85E69WF',\n",
       "  'number': 50158,\n",
       "  'title': 'CI: remove aiobotocore pin',\n",
       "  'user': {'login': 'jorisvandenbossche',\n",
       "   'id': 1020496,\n",
       "   'node_id': 'MDQ6VXNlcjEwMjA0OTY=',\n",
       "   'avatar_url': 'https://avatars.githubusercontent.com/u/1020496?v=4',\n",
       "   'gravatar_id': '',\n",
       "   'url': 'https://api.github.com/users/jorisvandenbossche',\n",
       "   'html_url': 'https://github.com/jorisvandenbossche',\n",
       "   'followers_url': 'https://api.github.com/users/jorisvandenbossche/followers',\n",
       "   'following_url': 'https://api.github.com/users/jorisvandenbossche/following{/other_user}',\n",
       "   'gists_url': 'https://api.github.com/users/jorisvandenbossche/gists{/gist_id}',\n",
       "   'starred_url': 'https://api.github.com/users/jorisvandenbossche/starred{/owner}{/repo}',\n",
       "   'subscriptions_url': 'https://api.github.com/users/jorisvandenbossche/subscriptions',\n",
       "   'organizations_url': 'https://api.github.com/users/jorisvandenbossche/orgs',\n",
       "   'repos_url': 'https://api.github.com/users/jorisvandenbossche/repos',\n",
       "   'events_url': 'https://api.github.com/users/jorisvandenbossche/events{/privacy}',\n",
       "   'received_events_url': 'https://api.github.com/users/jorisvandenbossche/received_events',\n",
       "   'type': 'User',\n",
       "   'site_admin': False},\n",
       "  'labels': [{'id': 48070600,\n",
       "    'node_id': 'MDU6TGFiZWw0ODA3MDYwMA==',\n",
       "    'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/CI',\n",
       "    'name': 'CI',\n",
       "    'color': 'a2bca7',\n",
       "    'default': False,\n",
       "    'description': 'Continuous Integration'}],\n",
       "  'state': 'open',\n",
       "  'locked': False,\n",
       "  'assignee': None,\n",
       "  'assignees': [],\n",
       "  'milestone': None,\n",
       "  'comments': 1,\n",
       "  'created_at': '2022-12-09T16:27:33Z',\n",
       "  'updated_at': '2022-12-10T20:07:08Z',\n",
       "  'closed_at': None,\n",
       "  'author_association': 'MEMBER',\n",
       "  'active_lock_reason': None,\n",
       "  'draft': False,\n",
       "  'pull_request': {'url': 'https://api.github.com/repos/pandas-dev/pandas/pulls/50158',\n",
       "   'html_url': 'https://github.com/pandas-dev/pandas/pull/50158',\n",
       "   'diff_url': 'https://github.com/pandas-dev/pandas/pull/50158.diff',\n",
       "   'patch_url': 'https://github.com/pandas-dev/pandas/pull/50158.patch',\n",
       "   'merged_at': None},\n",
       "  'body': 'Closes https://github.com/pandas-dev/pandas/issues/44311',\n",
       "  'reactions': {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50158/reactions',\n",
       "   'total_count': 0,\n",
       "   '+1': 0,\n",
       "   '-1': 0,\n",
       "   'laugh': 0,\n",
       "   'hooray': 0,\n",
       "   'confused': 0,\n",
       "   'heart': 0,\n",
       "   'rocket': 0,\n",
       "   'eyes': 0},\n",
       "  'timeline_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50158/timeline',\n",
       "  'performed_via_github_app': None,\n",
       "  'state_reason': None},\n",
       " {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50157',\n",
       "  'repository_url': 'https://api.github.com/repos/pandas-dev/pandas',\n",
       "  'labels_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50157/labels{/name}',\n",
       "  'comments_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50157/comments',\n",
       "  'events_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50157/events',\n",
       "  'html_url': 'https://github.com/pandas-dev/pandas/pull/50157',\n",
       "  'id': 1487033327,\n",
       "  'node_id': 'PR_kwDOAA0YD85E68is',\n",
       "  'number': 50157,\n",
       "  'title': 'DEV: remove downstream test packages from environment.yml',\n",
       "  'user': {'login': 'jorisvandenbossche',\n",
       "   'id': 1020496,\n",
       "   'node_id': 'MDQ6VXNlcjEwMjA0OTY=',\n",
       "   'avatar_url': 'https://avatars.githubusercontent.com/u/1020496?v=4',\n",
       "   'gravatar_id': '',\n",
       "   'url': 'https://api.github.com/users/jorisvandenbossche',\n",
       "   'html_url': 'https://github.com/jorisvandenbossche',\n",
       "   'followers_url': 'https://api.github.com/users/jorisvandenbossche/followers',\n",
       "   'following_url': 'https://api.github.com/users/jorisvandenbossche/following{/other_user}',\n",
       "   'gists_url': 'https://api.github.com/users/jorisvandenbossche/gists{/gist_id}',\n",
       "   'starred_url': 'https://api.github.com/users/jorisvandenbossche/starred{/owner}{/repo}',\n",
       "   'subscriptions_url': 'https://api.github.com/users/jorisvandenbossche/subscriptions',\n",
       "   'organizations_url': 'https://api.github.com/users/jorisvandenbossche/orgs',\n",
       "   'repos_url': 'https://api.github.com/users/jorisvandenbossche/repos',\n",
       "   'events_url': 'https://api.github.com/users/jorisvandenbossche/events{/privacy}',\n",
       "   'received_events_url': 'https://api.github.com/users/jorisvandenbossche/received_events',\n",
       "   'type': 'User',\n",
       "   'site_admin': False},\n",
       "  'labels': [],\n",
       "  'state': 'open',\n",
       "  'locked': False,\n",
       "  'assignee': None,\n",
       "  'assignees': [],\n",
       "  'milestone': None,\n",
       "  'comments': 2,\n",
       "  'created_at': '2022-12-09T16:24:58Z',\n",
       "  'updated_at': '2022-12-10T20:38:11Z',\n",
       "  'closed_at': None,\n",
       "  'author_association': 'MEMBER',\n",
       "  'active_lock_reason': None,\n",
       "  'draft': False,\n",
       "  'pull_request': {'url': 'https://api.github.com/repos/pandas-dev/pandas/pulls/50157',\n",
       "   'html_url': 'https://github.com/pandas-dev/pandas/pull/50157',\n",
       "   'diff_url': 'https://github.com/pandas-dev/pandas/pull/50157.diff',\n",
       "   'patch_url': 'https://github.com/pandas-dev/pandas/pull/50157.patch',\n",
       "   'merged_at': None},\n",
       "  'body': 'xref https://github.com/pandas-dev/pandas/issues/49998',\n",
       "  'reactions': {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50157/reactions',\n",
       "   'total_count': 0,\n",
       "   '+1': 0,\n",
       "   '-1': 0,\n",
       "   'laugh': 0,\n",
       "   'hooray': 0,\n",
       "   'confused': 0,\n",
       "   'heart': 0,\n",
       "   'rocket': 0,\n",
       "   'eyes': 0},\n",
       "  'timeline_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50157/timeline',\n",
       "  'performed_via_github_app': None,\n",
       "  'state_reason': None},\n",
       " {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50156',\n",
       "  'repository_url': 'https://api.github.com/repos/pandas-dev/pandas',\n",
       "  'labels_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50156/labels{/name}',\n",
       "  'comments_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50156/comments',\n",
       "  'events_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50156/events',\n",
       "  'html_url': 'https://github.com/pandas-dev/pandas/pull/50156',\n",
       "  'id': 1486999835,\n",
       "  'node_id': 'PR_kwDOAA0YD85E61Eb',\n",
       "  'number': 50156,\n",
       "  'title': 'TST: Added test for groupby dropping NA when grouping on columns',\n",
       "  'user': {'login': 'pbhoopala',\n",
       "   'id': 34802460,\n",
       "   'node_id': 'MDQ6VXNlcjM0ODAyNDYw',\n",
       "   'avatar_url': 'https://avatars.githubusercontent.com/u/34802460?v=4',\n",
       "   'gravatar_id': '',\n",
       "   'url': 'https://api.github.com/users/pbhoopala',\n",
       "   'html_url': 'https://github.com/pbhoopala',\n",
       "   'followers_url': 'https://api.github.com/users/pbhoopala/followers',\n",
       "   'following_url': 'https://api.github.com/users/pbhoopala/following{/other_user}',\n",
       "   'gists_url': 'https://api.github.com/users/pbhoopala/gists{/gist_id}',\n",
       "   'starred_url': 'https://api.github.com/users/pbhoopala/starred{/owner}{/repo}',\n",
       "   'subscriptions_url': 'https://api.github.com/users/pbhoopala/subscriptions',\n",
       "   'organizations_url': 'https://api.github.com/users/pbhoopala/orgs',\n",
       "   'repos_url': 'https://api.github.com/users/pbhoopala/repos',\n",
       "   'events_url': 'https://api.github.com/users/pbhoopala/events{/privacy}',\n",
       "   'received_events_url': 'https://api.github.com/users/pbhoopala/received_events',\n",
       "   'type': 'User',\n",
       "   'site_admin': False},\n",
       "  'labels': [],\n",
       "  'state': 'open',\n",
       "  'locked': False,\n",
       "  'assignee': None,\n",
       "  'assignees': [],\n",
       "  'milestone': None,\n",
       "  'comments': 0,\n",
       "  'created_at': '2022-12-09T16:03:08Z',\n",
       "  'updated_at': '2022-12-09T16:57:27Z',\n",
       "  'closed_at': None,\n",
       "  'author_association': 'NONE',\n",
       "  'active_lock_reason': None,\n",
       "  'draft': False,\n",
       "  'pull_request': {'url': 'https://api.github.com/repos/pandas-dev/pandas/pulls/50156',\n",
       "   'html_url': 'https://github.com/pandas-dev/pandas/pull/50156',\n",
       "   'diff_url': 'https://github.com/pandas-dev/pandas/pull/50156.diff',\n",
       "   'patch_url': 'https://github.com/pandas-dev/pandas/pull/50156.patch',\n",
       "   'merged_at': None},\n",
       "  'body': '- [ ] closes #21755\\r\\n- [X] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature\\r\\n- [X] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).\\r\\n- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.\\r\\n- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.\\r\\n',\n",
       "  'reactions': {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50156/reactions',\n",
       "   'total_count': 0,\n",
       "   '+1': 0,\n",
       "   '-1': 0,\n",
       "   'laugh': 0,\n",
       "   'hooray': 0,\n",
       "   'confused': 0,\n",
       "   'heart': 0,\n",
       "   'rocket': 0,\n",
       "   'eyes': 0},\n",
       "  'timeline_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50156/timeline',\n",
       "  'performed_via_github_app': None,\n",
       "  'state_reason': None},\n",
       " {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50155',\n",
       "  'repository_url': 'https://api.github.com/repos/pandas-dev/pandas',\n",
       "  'labels_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50155/labels{/name}',\n",
       "  'comments_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50155/comments',\n",
       "  'events_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50155/events',\n",
       "  'html_url': 'https://github.com/pandas-dev/pandas/pull/50155',\n",
       "  'id': 1486948350,\n",
       "  'node_id': 'PR_kwDOAA0YD85E6ppz',\n",
       "  'number': 50155,\n",
       "  'title': 'BUG: displaying NaN in complex numbers does not work',\n",
       "  'user': {'login': 'phofl',\n",
       "   'id': 61934744,\n",
       "   'node_id': 'MDQ6VXNlcjYxOTM0NzQ0',\n",
       "   'avatar_url': 'https://avatars.githubusercontent.com/u/61934744?v=4',\n",
       "   'gravatar_id': '',\n",
       "   'url': 'https://api.github.com/users/phofl',\n",
       "   'html_url': 'https://github.com/phofl',\n",
       "   'followers_url': 'https://api.github.com/users/phofl/followers',\n",
       "   'following_url': 'https://api.github.com/users/phofl/following{/other_user}',\n",
       "   'gists_url': 'https://api.github.com/users/phofl/gists{/gist_id}',\n",
       "   'starred_url': 'https://api.github.com/users/phofl/starred{/owner}{/repo}',\n",
       "   'subscriptions_url': 'https://api.github.com/users/phofl/subscriptions',\n",
       "   'organizations_url': 'https://api.github.com/users/phofl/orgs',\n",
       "   'repos_url': 'https://api.github.com/users/phofl/repos',\n",
       "   'events_url': 'https://api.github.com/users/phofl/events{/privacy}',\n",
       "   'received_events_url': 'https://api.github.com/users/phofl/received_events',\n",
       "   'type': 'User',\n",
       "   'site_admin': False},\n",
       "  'labels': [{'id': 13101118,\n",
       "    'node_id': 'MDU6TGFiZWwxMzEwMTExOA==',\n",
       "    'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Output-Formatting',\n",
       "    'name': 'Output-Formatting',\n",
       "    'color': 'ededed',\n",
       "    'default': False,\n",
       "    'description': '__repr__ of pandas objects, to_string'},\n",
       "   {'id': 172091424,\n",
       "    'node_id': 'MDU6TGFiZWwxNzIwOTE0MjQ=',\n",
       "    'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Complex',\n",
       "    'name': 'Complex',\n",
       "    'color': '009800',\n",
       "    'default': False,\n",
       "    'description': 'Complex Numbers'}],\n",
       "  'state': 'open',\n",
       "  'locked': False,\n",
       "  'assignee': None,\n",
       "  'assignees': [],\n",
       "  'milestone': None,\n",
       "  'comments': 0,\n",
       "  'created_at': '2022-12-09T15:28:24Z',\n",
       "  'updated_at': '2022-12-10T19:59:28Z',\n",
       "  'closed_at': None,\n",
       "  'author_association': 'MEMBER',\n",
       "  'active_lock_reason': None,\n",
       "  'draft': False,\n",
       "  'pull_request': {'url': 'https://api.github.com/repos/pandas-dev/pandas/pulls/50155',\n",
       "   'html_url': 'https://github.com/pandas-dev/pandas/pull/50155',\n",
       "   'diff_url': 'https://github.com/pandas-dev/pandas/pull/50155.diff',\n",
       "   'patch_url': 'https://github.com/pandas-dev/pandas/pull/50155.patch',\n",
       "   'merged_at': None},\n",
       "  'body': '- [x] closes #50069 (Replace xxxx with the GitHub issue number)\\r\\n- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature\\r\\n- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).\\r\\n- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.\\r\\n- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.\\r\\n\\r\\nNot sure if this is the best way to format things',\n",
       "  'reactions': {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50155/reactions',\n",
       "   'total_count': 0,\n",
       "   '+1': 0,\n",
       "   '-1': 0,\n",
       "   'laugh': 0,\n",
       "   'hooray': 0,\n",
       "   'confused': 0,\n",
       "   'heart': 0,\n",
       "   'rocket': 0,\n",
       "   'eyes': 0},\n",
       "  'timeline_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50155/timeline',\n",
       "  'performed_via_github_app': None,\n",
       "  'state_reason': None},\n",
       " {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50151',\n",
       "  'repository_url': 'https://api.github.com/repos/pandas-dev/pandas',\n",
       "  'labels_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50151/labels{/name}',\n",
       "  'comments_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50151/comments',\n",
       "  'events_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50151/events',\n",
       "  'html_url': 'https://github.com/pandas-dev/pandas/pull/50151',\n",
       "  'id': 1486854021,\n",
       "  'node_id': 'PR_kwDOAA0YD85E6Uu9',\n",
       "  'number': 50151,\n",
       "  'title': 'BUG: displaying string dtypes not showing storage option',\n",
       "  'user': {'login': 'phofl',\n",
       "   'id': 61934744,\n",
       "   'node_id': 'MDQ6VXNlcjYxOTM0NzQ0',\n",
       "   'avatar_url': 'https://avatars.githubusercontent.com/u/61934744?v=4',\n",
       "   'gravatar_id': '',\n",
       "   'url': 'https://api.github.com/users/phofl',\n",
       "   'html_url': 'https://github.com/phofl',\n",
       "   'followers_url': 'https://api.github.com/users/phofl/followers',\n",
       "   'following_url': 'https://api.github.com/users/phofl/following{/other_user}',\n",
       "   'gists_url': 'https://api.github.com/users/phofl/gists{/gist_id}',\n",
       "   'starred_url': 'https://api.github.com/users/phofl/starred{/owner}{/repo}',\n",
       "   'subscriptions_url': 'https://api.github.com/users/phofl/subscriptions',\n",
       "   'organizations_url': 'https://api.github.com/users/phofl/orgs',\n",
       "   'repos_url': 'https://api.github.com/users/phofl/repos',\n",
       "   'events_url': 'https://api.github.com/users/phofl/events{/privacy}',\n",
       "   'received_events_url': 'https://api.github.com/users/phofl/received_events',\n",
       "   'type': 'User',\n",
       "   'site_admin': False},\n",
       "  'labels': [{'id': 76811,\n",
       "    'node_id': 'MDU6TGFiZWw3NjgxMQ==',\n",
       "    'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Bug',\n",
       "    'name': 'Bug',\n",
       "    'color': 'e10c02',\n",
       "    'default': False,\n",
       "    'description': None},\n",
       "   {'id': 13101118,\n",
       "    'node_id': 'MDU6TGFiZWwxMzEwMTExOA==',\n",
       "    'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Output-Formatting',\n",
       "    'name': 'Output-Formatting',\n",
       "    'color': 'ededed',\n",
       "    'default': False,\n",
       "    'description': '__repr__ of pandas objects, to_string'},\n",
       "   {'id': 57522093,\n",
       "    'node_id': 'MDU6TGFiZWw1NzUyMjA5Mw==',\n",
       "    'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Strings',\n",
       "    'name': 'Strings',\n",
       "    'color': '5319e7',\n",
       "    'default': False,\n",
       "    'description': 'String extension data type and string data'}],\n",
       "  'state': 'open',\n",
       "  'locked': False,\n",
       "  'assignee': None,\n",
       "  'assignees': [],\n",
       "  'milestone': None,\n",
       "  'comments': 0,\n",
       "  'created_at': '2022-12-09T14:28:00Z',\n",
       "  'updated_at': '2022-12-10T19:59:11Z',\n",
       "  'closed_at': None,\n",
       "  'author_association': 'MEMBER',\n",
       "  'active_lock_reason': None,\n",
       "  'draft': False,\n",
       "  'pull_request': {'url': 'https://api.github.com/repos/pandas-dev/pandas/pulls/50151',\n",
       "   'html_url': 'https://github.com/pandas-dev/pandas/pull/50151',\n",
       "   'diff_url': 'https://github.com/pandas-dev/pandas/pull/50151.diff',\n",
       "   'patch_url': 'https://github.com/pandas-dev/pandas/pull/50151.patch',\n",
       "   'merged_at': None},\n",
       "  'body': '- [x] closes #50099 (Replace xxxx with the GitHub issue number)\\r\\n- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature\\r\\n- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).\\r\\n- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.\\r\\n- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.\\r\\n\\r\\ncc @mroeschke not saying that this is a good fix, but this shows that we have to use repr for the string dtypes in some way. Is there a more general rule to accomplish this?',\n",
       "  'reactions': {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50151/reactions',\n",
       "   'total_count': 0,\n",
       "   '+1': 0,\n",
       "   '-1': 0,\n",
       "   'laugh': 0,\n",
       "   'hooray': 0,\n",
       "   'confused': 0,\n",
       "   'heart': 0,\n",
       "   'rocket': 0,\n",
       "   'eyes': 0},\n",
       "  'timeline_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50151/timeline',\n",
       "  'performed_via_github_app': None,\n",
       "  'state_reason': None},\n",
       " {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50150',\n",
       "  'repository_url': 'https://api.github.com/repos/pandas-dev/pandas',\n",
       "  'labels_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50150/labels{/name}',\n",
       "  'comments_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50150/comments',\n",
       "  'events_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50150/events',\n",
       "  'html_url': 'https://github.com/pandas-dev/pandas/pull/50150',\n",
       "  'id': 1486815472,\n",
       "  'node_id': 'PR_kwDOAA0YD85E6MKr',\n",
       "  'number': 50150,\n",
       "  'title': 'ENH: improve error reporting for dup columns in merge_asof',\n",
       "  'user': {'login': 'phofl',\n",
       "   'id': 61934744,\n",
       "   'node_id': 'MDQ6VXNlcjYxOTM0NzQ0',\n",
       "   'avatar_url': 'https://avatars.githubusercontent.com/u/61934744?v=4',\n",
       "   'gravatar_id': '',\n",
       "   'url': 'https://api.github.com/users/phofl',\n",
       "   'html_url': 'https://github.com/phofl',\n",
       "   'followers_url': 'https://api.github.com/users/phofl/followers',\n",
       "   'following_url': 'https://api.github.com/users/phofl/following{/other_user}',\n",
       "   'gists_url': 'https://api.github.com/users/phofl/gists{/gist_id}',\n",
       "   'starred_url': 'https://api.github.com/users/phofl/starred{/owner}{/repo}',\n",
       "   'subscriptions_url': 'https://api.github.com/users/phofl/subscriptions',\n",
       "   'organizations_url': 'https://api.github.com/users/phofl/orgs',\n",
       "   'repos_url': 'https://api.github.com/users/phofl/repos',\n",
       "   'events_url': 'https://api.github.com/users/phofl/events{/privacy}',\n",
       "   'received_events_url': 'https://api.github.com/users/phofl/received_events',\n",
       "   'type': 'User',\n",
       "   'site_admin': False},\n",
       "  'labels': [{'id': 42670965,\n",
       "    'node_id': 'MDU6TGFiZWw0MjY3MDk2NQ==',\n",
       "    'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Error%20Reporting',\n",
       "    'name': 'Error Reporting',\n",
       "    'color': 'ffa0ff',\n",
       "    'default': False,\n",
       "    'description': 'Incorrect or improved errors from pandas'}],\n",
       "  'state': 'open',\n",
       "  'locked': False,\n",
       "  'assignee': None,\n",
       "  'assignees': [],\n",
       "  'milestone': None,\n",
       "  'comments': 0,\n",
       "  'created_at': '2022-12-09T14:03:09Z',\n",
       "  'updated_at': '2022-12-10T19:59:20Z',\n",
       "  'closed_at': None,\n",
       "  'author_association': 'MEMBER',\n",
       "  'active_lock_reason': None,\n",
       "  'draft': False,\n",
       "  'pull_request': {'url': 'https://api.github.com/repos/pandas-dev/pandas/pulls/50150',\n",
       "   'html_url': 'https://github.com/pandas-dev/pandas/pull/50150',\n",
       "   'diff_url': 'https://github.com/pandas-dev/pandas/pull/50150.diff',\n",
       "   'patch_url': 'https://github.com/pandas-dev/pandas/pull/50150.patch',\n",
       "   'merged_at': None},\n",
       "  'body': '- [x] closes #50102 (Replace xxxx with the GitHub issue number)\\r\\n- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature\\r\\n- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).\\r\\n- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.\\r\\n- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.\\r\\n',\n",
       "  'reactions': {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50150/reactions',\n",
       "   'total_count': 0,\n",
       "   '+1': 0,\n",
       "   '-1': 0,\n",
       "   'laugh': 0,\n",
       "   'hooray': 0,\n",
       "   'confused': 0,\n",
       "   'heart': 0,\n",
       "   'rocket': 0,\n",
       "   'eyes': 0},\n",
       "  'timeline_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50150/timeline',\n",
       "  'performed_via_github_app': None,\n",
       "  'state_reason': None},\n",
       " {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50149',\n",
       "  'repository_url': 'https://api.github.com/repos/pandas-dev/pandas',\n",
       "  'labels_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50149/labels{/name}',\n",
       "  'comments_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50149/comments',\n",
       "  'events_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50149/events',\n",
       "  'html_url': 'https://github.com/pandas-dev/pandas/pull/50149',\n",
       "  'id': 1486794913,\n",
       "  'node_id': 'PR_kwDOAA0YD85E6HkT',\n",
       "  'number': 50149,\n",
       "  'title': 'PERF: Speed up Period construction',\n",
       "  'user': {'login': 'lithomas1',\n",
       "   'id': 47963215,\n",
       "   'node_id': 'MDQ6VXNlcjQ3OTYzMjE1',\n",
       "   'avatar_url': 'https://avatars.githubusercontent.com/u/47963215?v=4',\n",
       "   'gravatar_id': '',\n",
       "   'url': 'https://api.github.com/users/lithomas1',\n",
       "   'html_url': 'https://github.com/lithomas1',\n",
       "   'followers_url': 'https://api.github.com/users/lithomas1/followers',\n",
       "   'following_url': 'https://api.github.com/users/lithomas1/following{/other_user}',\n",
       "   'gists_url': 'https://api.github.com/users/lithomas1/gists{/gist_id}',\n",
       "   'starred_url': 'https://api.github.com/users/lithomas1/starred{/owner}{/repo}',\n",
       "   'subscriptions_url': 'https://api.github.com/users/lithomas1/subscriptions',\n",
       "   'organizations_url': 'https://api.github.com/users/lithomas1/orgs',\n",
       "   'repos_url': 'https://api.github.com/users/lithomas1/repos',\n",
       "   'events_url': 'https://api.github.com/users/lithomas1/events{/privacy}',\n",
       "   'received_events_url': 'https://api.github.com/users/lithomas1/received_events',\n",
       "   'type': 'User',\n",
       "   'site_admin': False},\n",
       "  'labels': [{'id': 8935311,\n",
       "    'node_id': 'MDU6TGFiZWw4OTM1MzEx',\n",
       "    'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Performance',\n",
       "    'name': 'Performance',\n",
       "    'color': 'a10c02',\n",
       "    'default': False,\n",
       "    'description': 'Memory or execution speed performance'},\n",
       "   {'id': 60635328,\n",
       "    'node_id': 'MDU6TGFiZWw2MDYzNTMyOA==',\n",
       "    'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Period',\n",
       "    'name': 'Period',\n",
       "    'color': 'eb6420',\n",
       "    'default': False,\n",
       "    'description': 'Period data type'}],\n",
       "  'state': 'open',\n",
       "  'locked': False,\n",
       "  'assignee': None,\n",
       "  'assignees': [],\n",
       "  'milestone': None,\n",
       "  'comments': 0,\n",
       "  'created_at': '2022-12-09T13:49:33Z',\n",
       "  'updated_at': '2022-12-10T18:37:43Z',\n",
       "  'closed_at': None,\n",
       "  'author_association': 'MEMBER',\n",
       "  'active_lock_reason': None,\n",
       "  'draft': False,\n",
       "  'pull_request': {'url': 'https://api.github.com/repos/pandas-dev/pandas/pulls/50149',\n",
       "   'html_url': 'https://github.com/pandas-dev/pandas/pull/50149',\n",
       "   'diff_url': 'https://github.com/pandas-dev/pandas/pull/50149.diff',\n",
       "   'patch_url': 'https://github.com/pandas-dev/pandas/pull/50149.patch',\n",
       "   'merged_at': None},\n",
       "  'body': \"- [ ] closes #38312 (Replace xxxx with the GitHub issue number)\\r\\n- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature\\r\\n- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).\\r\\n- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.\\r\\n- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.\\r\\n\\r\\nASVs.\\r\\n```\\r\\n       before           after         ratio\\r\\n     [1d5ce5b3]       [b47aa45a]\\r\\n     <main>           <period-speedup>\\r\\n-     8.18±0.06ms       5.50±0.1ms     0.67  period.PeriodIndexConstructor.time_from_ints_daily('D', False)\\r\\n-     8.06±0.06ms       5.36±0.1ms     0.67  period.PeriodIndexConstructor.time_from_ints_daily('D', True)\\r\\n-     8.94±0.08ms      5.31±0.04ms     0.59  period.PeriodIndexConstructor.time_from_ints('D', False)\\r\\n-     8.77±0.09ms       5.20±0.1ms     0.59  period.PeriodIndexConstructor.time_from_ints('D', True)\\r\\n\\r\\nSOME BENCHMARKS HAVE CHANGED SIGNIFICANTLY.\\r\\nPERFORMANCE INCREASED.\\r\\n```\\r\\n\\r\\nparse_time_string already calls string_to_dts(parse_iso_8601_datetime), which I think is the only function that can currently parse nanoseconds(dateutil can't do it IIRC). \",\n",
       "  'reactions': {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50149/reactions',\n",
       "   'total_count': 0,\n",
       "   '+1': 0,\n",
       "   '-1': 0,\n",
       "   'laugh': 0,\n",
       "   'hooray': 0,\n",
       "   'confused': 0,\n",
       "   'heart': 0,\n",
       "   'rocket': 0,\n",
       "   'eyes': 0},\n",
       "  'timeline_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50149/timeline',\n",
       "  'performed_via_github_app': None,\n",
       "  'state_reason': None},\n",
       " {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50148',\n",
       "  'repository_url': 'https://api.github.com/repos/pandas-dev/pandas',\n",
       "  'labels_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50148/labels{/name}',\n",
       "  'comments_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50148/comments',\n",
       "  'events_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50148/events',\n",
       "  'html_url': 'https://github.com/pandas-dev/pandas/pull/50148',\n",
       "  'id': 1486602123,\n",
       "  'node_id': 'PR_kwDOAA0YD85E5ckb',\n",
       "  'number': 50148,\n",
       "  'title': 'PERF: Split blocks in blk.delete',\n",
       "  'user': {'login': 'phofl',\n",
       "   'id': 61934744,\n",
       "   'node_id': 'MDQ6VXNlcjYxOTM0NzQ0',\n",
       "   'avatar_url': 'https://avatars.githubusercontent.com/u/61934744?v=4',\n",
       "   'gravatar_id': '',\n",
       "   'url': 'https://api.github.com/users/phofl',\n",
       "   'html_url': 'https://github.com/phofl',\n",
       "   'followers_url': 'https://api.github.com/users/phofl/followers',\n",
       "   'following_url': 'https://api.github.com/users/phofl/following{/other_user}',\n",
       "   'gists_url': 'https://api.github.com/users/phofl/gists{/gist_id}',\n",
       "   'starred_url': 'https://api.github.com/users/phofl/starred{/owner}{/repo}',\n",
       "   'subscriptions_url': 'https://api.github.com/users/phofl/subscriptions',\n",
       "   'organizations_url': 'https://api.github.com/users/phofl/orgs',\n",
       "   'repos_url': 'https://api.github.com/users/phofl/repos',\n",
       "   'events_url': 'https://api.github.com/users/phofl/events{/privacy}',\n",
       "   'received_events_url': 'https://api.github.com/users/phofl/received_events',\n",
       "   'type': 'User',\n",
       "   'site_admin': False},\n",
       "  'labels': [],\n",
       "  'state': 'open',\n",
       "  'locked': False,\n",
       "  'assignee': None,\n",
       "  'assignees': [],\n",
       "  'milestone': None,\n",
       "  'comments': 2,\n",
       "  'created_at': '2022-12-09T11:36:36Z',\n",
       "  'updated_at': '2022-12-10T20:00:02Z',\n",
       "  'closed_at': None,\n",
       "  'author_association': 'MEMBER',\n",
       "  'active_lock_reason': None,\n",
       "  'draft': False,\n",
       "  'pull_request': {'url': 'https://api.github.com/repos/pandas-dev/pandas/pulls/50148',\n",
       "   'html_url': 'https://github.com/pandas-dev/pandas/pull/50148',\n",
       "   'diff_url': 'https://github.com/pandas-dev/pandas/pull/50148.diff',\n",
       "   'patch_url': 'https://github.com/pandas-dev/pandas/pull/50148.patch',\n",
       "   'merged_at': None},\n",
       "  'body': '- [x] closes #46267 (Replace xxxx with the GitHub issue number)\\r\\n- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature\\r\\n- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).\\r\\n- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.\\r\\n- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.\\r\\n\\r\\n```\\r\\n       before           after         ratio\\r\\n     [a047fb6c]       [e8c3a52a]\\r\\n-      1.16±0.01s         962±10μs     0.00  indexing.Setitem.time_setitem_list\\r\\n-         382±7ms         302±10μs     0.00  indexing.Setitem.time_setitem\\r\\n```',\n",
       "  'reactions': {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50148/reactions',\n",
       "   'total_count': 0,\n",
       "   '+1': 0,\n",
       "   '-1': 0,\n",
       "   'laugh': 0,\n",
       "   'hooray': 0,\n",
       "   'confused': 0,\n",
       "   'heart': 0,\n",
       "   'rocket': 0,\n",
       "   'eyes': 0},\n",
       "  'timeline_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50148/timeline',\n",
       "  'performed_via_github_app': None,\n",
       "  'state_reason': None},\n",
       " {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50147',\n",
       "  'repository_url': 'https://api.github.com/repos/pandas-dev/pandas',\n",
       "  'labels_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50147/labels{/name}',\n",
       "  'comments_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50147/comments',\n",
       "  'events_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50147/events',\n",
       "  'html_url': 'https://github.com/pandas-dev/pandas/pull/50147',\n",
       "  'id': 1486544837,\n",
       "  'node_id': 'PR_kwDOAA0YD85E5PyM',\n",
       "  'number': 50147,\n",
       "  'title': 'BUG: keep value types when calling .items() on Series',\n",
       "  'user': {'login': 'Kerybas',\n",
       "   'id': 42703200,\n",
       "   'node_id': 'MDQ6VXNlcjQyNzAzMjAw',\n",
       "   'avatar_url': 'https://avatars.githubusercontent.com/u/42703200?v=4',\n",
       "   'gravatar_id': '',\n",
       "   'url': 'https://api.github.com/users/Kerybas',\n",
       "   'html_url': 'https://github.com/Kerybas',\n",
       "   'followers_url': 'https://api.github.com/users/Kerybas/followers',\n",
       "   'following_url': 'https://api.github.com/users/Kerybas/following{/other_user}',\n",
       "   'gists_url': 'https://api.github.com/users/Kerybas/gists{/gist_id}',\n",
       "   'starred_url': 'https://api.github.com/users/Kerybas/starred{/owner}{/repo}',\n",
       "   'subscriptions_url': 'https://api.github.com/users/Kerybas/subscriptions',\n",
       "   'organizations_url': 'https://api.github.com/users/Kerybas/orgs',\n",
       "   'repos_url': 'https://api.github.com/users/Kerybas/repos',\n",
       "   'events_url': 'https://api.github.com/users/Kerybas/events{/privacy}',\n",
       "   'received_events_url': 'https://api.github.com/users/Kerybas/received_events',\n",
       "   'type': 'User',\n",
       "   'site_admin': False},\n",
       "  'labels': [],\n",
       "  'state': 'open',\n",
       "  'locked': False,\n",
       "  'assignee': None,\n",
       "  'assignees': [],\n",
       "  'milestone': None,\n",
       "  'comments': 4,\n",
       "  'created_at': '2022-12-09T10:58:26Z',\n",
       "  'updated_at': '2022-12-09T13:48:15Z',\n",
       "  'closed_at': None,\n",
       "  'author_association': 'NONE',\n",
       "  'active_lock_reason': None,\n",
       "  'draft': True,\n",
       "  'pull_request': {'url': 'https://api.github.com/repos/pandas-dev/pandas/pulls/50147',\n",
       "   'html_url': 'https://github.com/pandas-dev/pandas/pull/50147',\n",
       "   'diff_url': 'https://github.com/pandas-dev/pandas/pull/50147.diff',\n",
       "   'patch_url': 'https://github.com/pandas-dev/pandas/pull/50147.patch',\n",
       "   'merged_at': None},\n",
       "  'body': \"- [ ] closes #50125\\r\\n- [ ] closes #25969\\r\\n- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature\\r\\n- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).\\r\\n\\r\\nI have identified that the `.items()` method of `Series` modifies the type of values, before returning them.  \\r\\nOne of the major consequences (for me) is that the `.to_dict()` method unexpectedly returns modified types.  \\r\\nThis can be problematic with `float32` or `float16` types, as `.items()` would convert them to Python's float, thus changing the actual value of the output by adding some extra decimal digits.\\r\\n\\r\\n`.items()` modifies the type because it calls the `.__iter__()` method of the parent's class `base.IndexOpsMixin`, which [enforces the return of Python types](https://github.com/pandas-dev/pandas/blob/d1d8b666abdb03700703a39d6f9b48c5638d4d62/pandas/core/base.py#L750).\\r\\n\\r\\nFrom my point, `.items()` and `.to_dict()` should not change the type of the values. This can be easily done by refering to `Series._values`, as proposed in this PR.\",\n",
       "  'reactions': {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50147/reactions',\n",
       "   'total_count': 0,\n",
       "   '+1': 0,\n",
       "   '-1': 0,\n",
       "   'laugh': 0,\n",
       "   'hooray': 0,\n",
       "   'confused': 0,\n",
       "   'heart': 0,\n",
       "   'rocket': 0,\n",
       "   'eyes': 0},\n",
       "  'timeline_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50147/timeline',\n",
       "  'performed_via_github_app': None,\n",
       "  'state_reason': None},\n",
       " {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50145',\n",
       "  'repository_url': 'https://api.github.com/repos/pandas-dev/pandas',\n",
       "  'labels_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50145/labels{/name}',\n",
       "  'comments_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50145/comments',\n",
       "  'events_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50145/events',\n",
       "  'html_url': 'https://github.com/pandas-dev/pandas/pull/50145',\n",
       "  'id': 1486402080,\n",
       "  'node_id': 'PR_kwDOAA0YD85E4wIL',\n",
       "  'number': 50145,\n",
       "  'title': 'DOC restructure contributing environment guide',\n",
       "  'user': {'login': 'MarcoGorelli',\n",
       "   'id': 33491632,\n",
       "   'node_id': 'MDQ6VXNlcjMzNDkxNjMy',\n",
       "   'avatar_url': 'https://avatars.githubusercontent.com/u/33491632?v=4',\n",
       "   'gravatar_id': '',\n",
       "   'url': 'https://api.github.com/users/MarcoGorelli',\n",
       "   'html_url': 'https://github.com/MarcoGorelli',\n",
       "   'followers_url': 'https://api.github.com/users/MarcoGorelli/followers',\n",
       "   'following_url': 'https://api.github.com/users/MarcoGorelli/following{/other_user}',\n",
       "   'gists_url': 'https://api.github.com/users/MarcoGorelli/gists{/gist_id}',\n",
       "   'starred_url': 'https://api.github.com/users/MarcoGorelli/starred{/owner}{/repo}',\n",
       "   'subscriptions_url': 'https://api.github.com/users/MarcoGorelli/subscriptions',\n",
       "   'organizations_url': 'https://api.github.com/users/MarcoGorelli/orgs',\n",
       "   'repos_url': 'https://api.github.com/users/MarcoGorelli/repos',\n",
       "   'events_url': 'https://api.github.com/users/MarcoGorelli/events{/privacy}',\n",
       "   'received_events_url': 'https://api.github.com/users/MarcoGorelli/received_events',\n",
       "   'type': 'User',\n",
       "   'site_admin': False},\n",
       "  'labels': [],\n",
       "  'state': 'open',\n",
       "  'locked': False,\n",
       "  'assignee': None,\n",
       "  'assignees': [],\n",
       "  'milestone': None,\n",
       "  'comments': 1,\n",
       "  'created_at': '2022-12-09T09:23:28Z',\n",
       "  'updated_at': '2022-12-09T09:29:49Z',\n",
       "  'closed_at': None,\n",
       "  'author_association': 'MEMBER',\n",
       "  'active_lock_reason': None,\n",
       "  'draft': True,\n",
       "  'pull_request': {'url': 'https://api.github.com/repos/pandas-dev/pandas/pulls/50145',\n",
       "   'html_url': 'https://github.com/pandas-dev/pandas/pull/50145',\n",
       "   'diff_url': 'https://github.com/pandas-dev/pandas/pull/50145.diff',\n",
       "   'patch_url': 'https://github.com/pandas-dev/pandas/pull/50145.patch',\n",
       "   'merged_at': None},\n",
       "  'body': \"- [ ] closes #50134 (Replace xxxx with the GitHub issue number)\\r\\n- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature\\r\\n- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).\\r\\n- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.\\r\\n- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.\\r\\n\\r\\nHey @noatamir , thoughts on this? I think it's clearer to split it up into 3 steps:\\r\\n\\r\\n![image](https://user-images.githubusercontent.com/33491632/206668601-02e3dfb7-0d3e-4093-8654-f295535e03ff.png)\\r\\n\\r\\nI've also removed some parts which are arguably unnecessary\",\n",
       "  'reactions': {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50145/reactions',\n",
       "   'total_count': 1,\n",
       "   '+1': 0,\n",
       "   '-1': 0,\n",
       "   'laugh': 0,\n",
       "   'hooray': 0,\n",
       "   'confused': 0,\n",
       "   'heart': 1,\n",
       "   'rocket': 0,\n",
       "   'eyes': 0},\n",
       "  'timeline_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50145/timeline',\n",
       "  'performed_via_github_app': None,\n",
       "  'state_reason': None},\n",
       " {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50143',\n",
       "  'repository_url': 'https://api.github.com/repos/pandas-dev/pandas',\n",
       "  'labels_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50143/labels{/name}',\n",
       "  'comments_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50143/comments',\n",
       "  'events_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50143/events',\n",
       "  'html_url': 'https://github.com/pandas-dev/pandas/issues/50143',\n",
       "  'id': 1486329126,\n",
       "  'node_id': 'I_kwDOAA0YD85Yl5Um',\n",
       "  'number': 50143,\n",
       "  'title': 'BUG: Error concatenating timezone-aware datetime column and object column containing only None',\n",
       "  'user': {'login': 'getuem',\n",
       "   'id': 50838995,\n",
       "   'node_id': 'MDQ6VXNlcjUwODM4OTk1',\n",
       "   'avatar_url': 'https://avatars.githubusercontent.com/u/50838995?v=4',\n",
       "   'gravatar_id': '',\n",
       "   'url': 'https://api.github.com/users/getuem',\n",
       "   'html_url': 'https://github.com/getuem',\n",
       "   'followers_url': 'https://api.github.com/users/getuem/followers',\n",
       "   'following_url': 'https://api.github.com/users/getuem/following{/other_user}',\n",
       "   'gists_url': 'https://api.github.com/users/getuem/gists{/gist_id}',\n",
       "   'starred_url': 'https://api.github.com/users/getuem/starred{/owner}{/repo}',\n",
       "   'subscriptions_url': 'https://api.github.com/users/getuem/subscriptions',\n",
       "   'organizations_url': 'https://api.github.com/users/getuem/orgs',\n",
       "   'repos_url': 'https://api.github.com/users/getuem/repos',\n",
       "   'events_url': 'https://api.github.com/users/getuem/events{/privacy}',\n",
       "   'received_events_url': 'https://api.github.com/users/getuem/received_events',\n",
       "   'type': 'User',\n",
       "   'site_admin': False},\n",
       "  'labels': [{'id': 76811,\n",
       "    'node_id': 'MDU6TGFiZWw3NjgxMQ==',\n",
       "    'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Bug',\n",
       "    'name': 'Bug',\n",
       "    'color': 'e10c02',\n",
       "    'default': False,\n",
       "    'description': None},\n",
       "   {'id': 1954720290,\n",
       "    'node_id': 'MDU6TGFiZWwxOTU0NzIwMjkw',\n",
       "    'url': 'https://api.github.com/repos/pandas-dev/pandas/labels/Needs%20Triage',\n",
       "    'name': 'Needs Triage',\n",
       "    'color': '0052cc',\n",
       "    'default': False,\n",
       "    'description': 'Issue that has not been reviewed by a pandas team member'}],\n",
       "  'state': 'open',\n",
       "  'locked': False,\n",
       "  'assignee': None,\n",
       "  'assignees': [],\n",
       "  'milestone': None,\n",
       "  'comments': 0,\n",
       "  'created_at': '2022-12-09T08:38:54Z',\n",
       "  'updated_at': '2022-12-09T08:53:06Z',\n",
       "  'closed_at': None,\n",
       "  'author_association': 'NONE',\n",
       "  'active_lock_reason': None,\n",
       "  'body': '### Pandas version checks\\r\\n\\r\\n- [X] I have checked that this issue has not already been reported.\\r\\n\\r\\n- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.\\r\\n\\r\\n- [ ] I have confirmed this bug exists on the main branch of pandas.\\r\\n\\r\\n\\r\\n### Reproducible Example\\r\\n\\r\\n```python\\r\\nimport pandas as pd\\r\\nimport pytz\\r\\nfrom datetime import datetime\\r\\n\\r\\ndf1 = pd.DataFrame([{\\'date\\': datetime(2022, 12, 1, 12, 0, 0, 0, pytz.UTC)}])\\r\\ndf2 = pd.DataFrame([{\\'date\\': None}])\\r\\npd.concat([df1, df2])\\r\\n```\\r\\n\\r\\n### Traceback\\r\\n\\r\\n<details>\\r\\n\\r\\n```python-traceback\\r\\n\\r\\n---------------------------------------------------------------------------\\r\\nAttributeError                            Traceback (most recent call last)\\r\\nInput In [55], in <cell line: 3>()\\r\\n      1 df1 = pd.DataFrame([{\\'date\\': datetime(2022, 12, 1, 12, 0, 0, 0, pytz.UTC)}])\\r\\n      2 df2 = pd.DataFrame([{\\'date\\': None}])\\r\\n----> 3 pd.concat([df1, df2])\\r\\n\\r\\nFile ~/software/mambaforge/envs/dev/lib/python3.10/site-packages/pandas/util/_decorators.py:331, in deprecate_nonkeyword_arguments.<locals>.decorate.<locals>.wrapper(*args, **kwargs)\\r\\n    325 if len(args) > num_allow_args:\\r\\n    326     warnings.warn(\\r\\n    327         msg.format(arguments=_format_argument_list(allow_args)),\\r\\n    328         FutureWarning,\\r\\n    329         stacklevel=find_stack_level(),\\r\\n    330     )\\r\\n--> 331 return func(*args, **kwargs)\\r\\n\\r\\nFile ~/software/mambaforge/envs/dev/lib/python3.10/site-packages/pandas/core/reshape/concat.py:381, in concat(objs, axis, join, ignore_index, keys, levels, names, verify_integrity, sort, copy)\\r\\n    159 \"\"\"\\r\\n    160 Concatenate pandas objects along a particular axis.\\r\\n    161 \\r\\n   (...)\\r\\n    366 1   3   4\\r\\n    367 \"\"\"\\r\\n    368 op = _Concatenator(\\r\\n    369     objs,\\r\\n    370     axis=axis,\\r\\n   (...)\\r\\n    378     sort=sort,\\r\\n    379 )\\r\\n--> 381 return op.get_result()\\r\\n\\r\\nFile ~/software/mambaforge/envs/dev/lib/python3.10/site-packages/pandas/core/reshape/concat.py:616, in _Concatenator.get_result(self)\\r\\n    612             indexers[ax] = obj_labels.get_indexer(new_labels)\\r\\n    614     mgrs_indexers.append((obj._mgr, indexers))\\r\\n--> 616 new_data = concatenate_managers(\\r\\n    617     mgrs_indexers, self.new_axes, concat_axis=self.bm_axis, copy=self.copy\\r\\n    618 )\\r\\n    619 if not self.copy:\\r\\n    620     new_data._consolidate_inplace()\\r\\n\\r\\nFile ~/software/mambaforge/envs/dev/lib/python3.10/site-packages/pandas/core/internals/concat.py:233, in concatenate_managers(mgrs_indexers, axes, concat_axis, copy)\\r\\n    231     fastpath = blk.values.dtype == values.dtype\\r\\n    232 else:\\r\\n--> 233     values = _concatenate_join_units(join_units, concat_axis, copy=copy)\\r\\n    234     fastpath = False\\r\\n    236 if fastpath:\\r\\n\\r\\nFile ~/software/mambaforge/envs/dev/lib/python3.10/site-packages/pandas/core/internals/concat.py:542, in _concatenate_join_units(join_units, concat_axis, copy)\\r\\n    539 has_none_blocks = any(unit.block.dtype.kind == \"V\" for unit in join_units)\\r\\n    540 upcasted_na = _dtype_to_na_value(empty_dtype, has_none_blocks)\\r\\n--> 542 to_concat = [\\r\\n    543     ju.get_reindexed_values(empty_dtype=empty_dtype, upcasted_na=upcasted_na)\\r\\n    544     for ju in join_units\\r\\n    545 ]\\r\\n    547 if len(to_concat) == 1:\\r\\n    548     # Only one block, nothing to concatenate.\\r\\n    549     concat_values = to_concat[0]\\r\\n\\r\\nFile ~/software/mambaforge/envs/dev/lib/python3.10/site-packages/pandas/core/internals/concat.py:543, in <listcomp>(.0)\\r\\n    539 has_none_blocks = any(unit.block.dtype.kind == \"V\" for unit in join_units)\\r\\n    540 upcasted_na = _dtype_to_na_value(empty_dtype, has_none_blocks)\\r\\n    542 to_concat = [\\r\\n--> 543     ju.get_reindexed_values(empty_dtype=empty_dtype, upcasted_na=upcasted_na)\\r\\n    544     for ju in join_units\\r\\n    545 ]\\r\\n    547 if len(to_concat) == 1:\\r\\n    548     # Only one block, nothing to concatenate.\\r\\n    549     concat_values = to_concat[0]\\r\\n\\r\\nFile ~/software/mambaforge/envs/dev/lib/python3.10/site-packages/pandas/core/internals/concat.py:468, in JoinUnit.get_reindexed_values(self, empty_dtype, upcasted_na)\\r\\n    464         fill_value = None\\r\\n    466 if isinstance(empty_dtype, DatetimeTZDtype):\\r\\n    467     # NB: exclude e.g. pyarrow[dt64tz] dtypes\\r\\n--> 468     i8values = np.full(self.shape, fill_value.value)\\r\\n    469     return DatetimeArray(i8values, dtype=empty_dtype)\\r\\n    471 elif is_1d_only_ea_dtype(empty_dtype):\\r\\n\\r\\nAttributeError: \\'NoneType\\' object has no attribute \\'value\\'\\r\\n\\r\\n```\\r\\n\\r\\n</details>\\r\\n\\r\\n### Issue Description\\r\\n\\r\\nConcatenating DataFrames fails with an AttributeError if attempting to concatenate a column with a timezone-aware datetime dtype and a column with the object dtype containing only `None` as values.\\r\\n\\r\\n### Expected Behavior\\r\\n\\r\\nThe expected behavior is for this to work as it does with timezone-naive datetimes:\\r\\n\\r\\n```python\\r\\ndf1 = pd.DataFrame([{\\'date\\': datetime(2022, 12, 9, 12, 0, 0, 0)}])\\r\\ndf2 = pd.DataFrame([{\\'date\\': None}])\\r\\npd.concat([df1, df2])\\r\\n```\\r\\n\\r\\n```python\\r\\n\\r\\n                 date\\r\\n0 2022-12-09 12:00:00\\r\\n0                 NaT\\r\\n```\\r\\n\\r\\n### Installed Versions\\r\\n\\r\\n<details>\\r\\n\\r\\nINSTALLED VERSIONS\\r\\n------------------\\r\\ncommit           : 8dab54d6573f7186ff0c3b6364d5e4dd635ff3e7\\r\\npython           : 3.10.6.final.0\\r\\npython-bits      : 64\\r\\nOS               : Linux\\r\\nOS-release       : 4.4.0-19041-Microsoft\\r\\nVersion          : #1237-Microsoft Sat Sep 11 14:32:00 PST 2021\\r\\nmachine          : x86_64\\r\\nprocessor        : x86_64\\r\\nbyteorder        : little\\r\\nLC_ALL           : None\\r\\nLANG             : C.UTF-8\\r\\nLOCALE           : en_US.UTF-8\\r\\n\\r\\npandas           : 1.5.2\\r\\nnumpy            : 1.23.2\\r\\npytz             : 2022.2.1\\r\\ndateutil         : 2.8.2\\r\\nsetuptools       : 65.3.0\\r\\npip              : 22.2.2\\r\\nCython           : None\\r\\npytest           : None\\r\\nhypothesis       : None\\r\\nsphinx           : None\\r\\nblosc            : None\\r\\nfeather          : None\\r\\nxlsxwriter       : None\\r\\nlxml.etree       : 4.8.0\\r\\nhtml5lib         : None\\r\\npymysql          : None\\r\\npsycopg2         : 2.9.3\\r\\njinja2           : 3.1.2\\r\\nIPython          : 8.4.0\\r\\npandas_datareader: None\\r\\nbs4              : 4.11.1\\r\\nbottleneck       : None\\r\\nbrotli           : \\r\\nfastparquet      : None\\r\\nfsspec           : None\\r\\ngcsfs            : None\\r\\nmatplotlib       : 3.5.3\\r\\nnumba            : None\\r\\nnumexpr          : None\\r\\nodfpy            : None\\r\\nopenpyxl         : 3.0.9\\r\\npandas_gbq       : None\\r\\npyarrow          : None\\r\\npyreadstat       : None\\r\\npyxlsb           : None\\r\\ns3fs             : None\\r\\nscipy            : 1.9.3\\r\\nsnappy           : None\\r\\nsqlalchemy       : None\\r\\ntables           : None\\r\\ntabulate         : None\\r\\nxarray           : None\\r\\nxlrd             : 2.0.1\\r\\nxlwt             : None\\r\\nzstandard        : None\\r\\ntzdata           : 2022.5\\r\\n\\r\\n</details>\\r\\n',\n",
       "  'reactions': {'url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50143/reactions',\n",
       "   'total_count': 0,\n",
       "   '+1': 0,\n",
       "   '-1': 0,\n",
       "   'laugh': 0,\n",
       "   'hooray': 0,\n",
       "   'confused': 0,\n",
       "   'heart': 0,\n",
       "   'rocket': 0,\n",
       "   'eyes': 0},\n",
       "  'timeline_url': 'https://api.github.com/repos/pandas-dev/pandas/issues/50143/timeline',\n",
       "  'performed_via_github_app': None,\n",
       "  'state_reason': None}]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "req = requests.get('https://api.github.com/repos/pandas-dev/pandas/issues')\n",
    "\n",
    "data = req.json()\n",
    "\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "fix some dosctring-RT02-error\n",
      " lightfm installation problem\n",
      "BUG: fixed issue where Index was being accessed with DatetimeIndex attribute\n",
      "DEPR: deprecate Index.is_integer\n",
      "BUG: ensure connection does not close before retrieving results from io.sql.execute.\n",
      "DEPR: deprecate Index.is_boolean\n",
      "BUG: isetitem coercing df columns to object\n",
      "Fixed issue with writing Decimal type to excel file - Issue #49598\n",
      "BUG: df.setitem raising for ea bool df indexer\n",
      "BUG: iloc raising for ea series\n",
      "BUG: loc for ea index raising with oob slice end\n",
      "Document generating compile_commands.json for C extensions\n",
      "BUG: reindex __cinit__() got multiple values for keyword argument 'verify_integrity'\n",
      "BUG: loc raising error for MultiIndex with bool indexer _and first level of object dtype_ \n",
      "TYP #37715: Fix mypy errors in column.py\n",
      "BUG: Deprecation warning for bool/numeric concatenation seems too broadly scoped\n",
      "BUG: In master, math with periodindex + index of offset produces wrong result\n",
      "BUG: `Series` slicing can fail for extension array `Index`\n",
      "Test ruff for linting\n",
      "CI: remove aiobotocore pin\n",
      "DEV: remove downstream test packages from environment.yml\n",
      "TST: Added test for groupby dropping NA when grouping on columns\n",
      "BUG: displaying NaN in complex numbers does not work\n",
      "BUG: displaying string dtypes not showing storage option\n",
      "ENH: improve error reporting for dup columns in merge_asof\n",
      "PERF: Speed up Period construction\n",
      "PERF: Split blocks in blk.delete\n",
      "BUG: keep value types when calling .items() on Series\n",
      "DOC restructure contributing environment guide\n",
      "BUG: Error concatenating timezone-aware datetime column and object column containing only None\n"
     ]
    }
   ],
   "source": [
    "for i in range(len(data)):\n",
    "    print(data[i]['title'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.10.6 64-bit",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.6"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "916dbcbb3f70747c44a77c7bcd40155683ae19c65e1c03b4aa3499c5328201f1"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
